<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.353">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Jiratchaya Nuanpirom">
<meta name="author" content="Prasert Yodsawat">
<meta name="author" content="Khunanon Chanasongkram">
<meta name="dcterms.date" content="2023-05-17">

<title>R Workshop 2023 - Plotting Omics Data</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<link href="../img/fav_abbl_logo.png" rel="icon" type="image/png">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-text-highlighting-styles">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-bootstrap" data-mode="light">
<link href="../site_libs/bootstrap/bootstrap-dark.min.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-bootstrap" data-mode="dark">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>

<script src="../site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="../site_libs/lightable-0.0.1/lightable.css" rel="stylesheet">


<link rel="stylesheet" href="../styles.css">
<meta property="og:title" content="R Workshop 2023 - Plotting Omics Data">
<meta property="og:description" content="">
<meta property="og:image" content="plotting_omics_data_files/figure-html/unnamed-chunk-2-1.png">
<meta property="og:site-name" content="R Workshop 2023">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a href="../index.html" class="navbar-brand navbar-brand-logo">
    <img src="../img/fav_abbl_logo.png" alt="" class="navbar-logo">
    </a>
    <a class="navbar-brand" href="../index.html">
    <span class="navbar-title">R Workshop 2023</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="../index.html" rel="" target="">
 <span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../schedule.html" rel="" target="">
 <span class="menu-text">Schedule</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../prerequisite.html" rel="" target="">
 <span class="menu-text">Prerequisites</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../about.html" rel="" target="">
 <span class="menu-text">About</span></a>
  </li>  
</ul>
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/JirathNuan/r-handviz-workshop" rel="" target=""><i class="bi bi-github" role="img">
</i> 
 <span class="menu-text"></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://www.facebook.com/pslabpsu" rel="" target=""><i class="bi bi-facebook" role="img">
</i> 
 <span class="menu-text"></span></a>
  </li>  
</ul>
            <div class="quarto-navbar-tools">
  <a href="" class="quarto-color-scheme-toggle quarto-navigation-tool  px-1" onclick="window.quartoToggleColorScheme(); return false;" title="Toggle dark mode"><i class="bi"></i></a>
  <a href="" class="quarto-reader-toggle quarto-navigation-tool px-1" onclick="window.quartoToggleReader(); return false;" title="Toggle reader mode">
  <div class="quarto-reader-toggle-btn">
  <i class="bi"></i>
  </div>
</a>
</div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#load-libraries" id="toc-load-libraries" class="nav-link active" data-scroll-target="#load-libraries"><span class="header-section-number">1</span> Load libraries</a></li>
  <li><a href="#plotting-abundance-of-representative-terms" id="toc-plotting-abundance-of-representative-terms" class="nav-link" data-scroll-target="#plotting-abundance-of-representative-terms"><span class="header-section-number">2</span> Plotting Abundance of Representative Terms</a></li>
  <li><a href="#principal-component-analysis" id="toc-principal-component-analysis" class="nav-link" data-scroll-target="#principal-component-analysis"><span class="header-section-number">3</span> Principal Component Analysis</a></li>
  <li><a href="#hierarchical-clustering-analysis-heat-maps" id="toc-hierarchical-clustering-analysis-heat-maps" class="nav-link" data-scroll-target="#hierarchical-clustering-analysis-heat-maps"><span class="header-section-number">4</span> Hierarchical Clustering Analysis (Heat maps)</a></li>
  <li><a href="#volcano-plot" id="toc-volcano-plot" class="nav-link" data-scroll-target="#volcano-plot"><span class="header-section-number">5</span> Volcano Plot</a></li>
  <li><a href="#venn-diagram-and-upset-plot" id="toc-venn-diagram-and-upset-plot" class="nav-link" data-scroll-target="#venn-diagram-and-upset-plot"><span class="header-section-number">6</span> Venn Diagram and Upset Plot</a>
  <ul class="collapse">
  <li><a href="#plotting-venn-diagram-with-ggvenn" id="toc-plotting-venn-diagram-with-ggvenn" class="nav-link" data-scroll-target="#plotting-venn-diagram-with-ggvenn"><span class="header-section-number">6.1</span> Plotting Venn diagram with <code>ggvenn</code></a></li>
  <li><a href="#venn-diagram-alternative-upset-plot" id="toc-venn-diagram-alternative-upset-plot" class="nav-link" data-scroll-target="#venn-diagram-alternative-upset-plot"><span class="header-section-number">6.2</span> Venn diagram alternative: Upset Plot</a></li>
  </ul></li>
  <li><a href="#ggplot2-based-publication-ready-plots-with-ggpubr" id="toc-ggplot2-based-publication-ready-plots-with-ggpubr" class="nav-link" data-scroll-target="#ggplot2-based-publication-ready-plots-with-ggpubr"><span class="header-section-number">7</span> ggplot2 Based Publication Ready Plots with <code>ggpubr</code></a>
  <ul class="collapse">
  <li><a href="#histogram" id="toc-histogram" class="nav-link" data-scroll-target="#histogram"><span class="header-section-number">7.1</span> Histogram</a></li>
  <li><a href="#density-plot" id="toc-density-plot" class="nav-link" data-scroll-target="#density-plot"><span class="header-section-number">7.2</span> Density plot</a></li>
  <li><a href="#simple-box-plot" id="toc-simple-box-plot" class="nav-link" data-scroll-target="#simple-box-plot"><span class="header-section-number">7.3</span> Simple box plot</a></li>
  <li><a href="#bar-and-line-plots-one-grouping-variable" id="toc-bar-and-line-plots-one-grouping-variable" class="nav-link" data-scroll-target="#bar-and-line-plots-one-grouping-variable"><span class="header-section-number">7.4</span> Bar and line plots (one grouping variable):</a></li>
  <li><a href="#scatter-plot" id="toc-scatter-plot" class="nav-link" data-scroll-target="#scatter-plot"><span class="header-section-number">7.5</span> Scatter plot</a></li>
  </ul></li>
  <li><a href="#session-info" id="toc-session-info" class="nav-link" data-scroll-target="#session-info"><span class="header-section-number">8</span> Session info</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Plotting Omics Data</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Authors</div>
    <div class="quarto-title-meta-contents">
             <p>Jiratchaya Nuanpirom </p>
             <p>Prasert Yodsawat </p>
             <p>Khunanon Chanasongkram </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">May 17, 2023</p>
    </div>
  </div>
  
    
  </div>
  

</header>

<section id="load-libraries" class="level2" data-number="1">
<h2 data-number="1" class="anchored" data-anchor-id="load-libraries"><span class="header-section-number">1</span> Load libraries</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a>pacman<span class="sc">::</span><span class="fu">p_load</span>(</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>  <span class="st">"tidyverse"</span>,</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a>  <span class="st">"kableExtra"</span>,</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a>  <span class="st">"openxlsx"</span>,</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a>  <span class="st">"DESeq2"</span>,</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>  <span class="st">"pcaExplorer"</span>,</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a>  <span class="st">"factoextra"</span>,</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a>  <span class="st">"ggsci"</span>,</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a>  <span class="st">"ComplexHeatmap"</span>,</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a>  <span class="st">"RColorBrewer"</span>,</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a>  <span class="st">"EnhancedVolcano"</span>,</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a>  <span class="st">"ggvenn"</span>, </span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a>  <span class="st">"ggpubr"</span>,</span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a>  <span class="st">"kableExtra"</span></span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="plotting-abundance-of-representative-terms" class="level2" data-number="2">
<h2 data-number="2" class="anchored" data-anchor-id="plotting-abundance-of-representative-terms"><span class="header-section-number">2</span> Plotting Abundance of Representative Terms</h2>
<div class="cell">
<div class="sourceCode cell-code" id="annotated-cell-2"><pre class="sourceCode r code-annotation-code code-with-copy code-annotated"><code class="sourceCode r"><span id="annotated-cell-2-1"><a href="#annotated-cell-2-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load COG dictionary</span></span>
<button class="code-annotation-anchor" data-target-cell="annotated-cell-2" data-target-annotation="1">1</button><span id="annotated-cell-2-2" class="code-annotation-target"><a href="#annotated-cell-2-2" aria-hidden="true" tabindex="-1"></a>COG_dict <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(<span class="st">"https://github.com/JirathNuan/r-handviz-workshop/raw/main/datasets/cog_category.txt"</span>)</span>
<span id="annotated-cell-2-3"><a href="#annotated-cell-2-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="annotated-cell-2-4"><a href="#annotated-cell-2-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Load eggNOG-mapper result</span></span>
<button class="code-annotation-anchor" data-target-cell="annotated-cell-2" data-target-annotation="2">2</button><span id="annotated-cell-2-5" class="code-annotation-target"><a href="#annotated-cell-2-5" aria-hidden="true" tabindex="-1"></a>emapper_dt <span class="ot">&lt;-</span> <span class="fu">read.xlsx</span>(<span class="st">"https://github.com/JirathNuan/r-handviz-workshop/raw/main/datasets/Dme_chr4.emapper.annotations.xlsx"</span>,</span>
<span id="annotated-cell-2-6"><a href="#annotated-cell-2-6" aria-hidden="true" tabindex="-1"></a>                        <span class="at">startRow =</span> <span class="dv">3</span>, </span>
<span id="annotated-cell-2-7"><a href="#annotated-cell-2-7" aria-hidden="true" tabindex="-1"></a>                        <span class="at">cols =</span> <span class="dv">7</span>) <span class="sc">%&gt;%</span> </span>
<button class="code-annotation-anchor" data-target-cell="annotated-cell-2" data-target-annotation="3">3</button><span id="annotated-cell-2-8" class="code-annotation-target"><a href="#annotated-cell-2-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(COG_category <span class="sc">!=</span> <span class="st">"-"</span>) <span class="sc">%&gt;%</span></span>
<button class="code-annotation-anchor" data-target-cell="annotated-cell-2" data-target-annotation="4">4</button><span id="annotated-cell-2-9" class="code-annotation-target"><a href="#annotated-cell-2-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(COG_category) <span class="sc">%&gt;%</span></span>
<button class="code-annotation-anchor" data-target-cell="annotated-cell-2" data-target-annotation="5">5</button><span id="annotated-cell-2-10" class="code-annotation-target"><a href="#annotated-cell-2-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span></span>
<button class="code-annotation-anchor" data-target-cell="annotated-cell-2" data-target-annotation="6">6</button><span id="annotated-cell-2-11" class="code-annotation-target"><a href="#annotated-cell-2-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(COG_dict, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"COG_category"</span> <span class="ot">=</span> <span class="st">"category"</span>)) <span class="sc">%&gt;%</span></span>
<button class="code-annotation-anchor" data-target-cell="annotated-cell-2" data-target-annotation="7">7</button><span id="annotated-cell-2-12" class="code-annotation-target"><a href="#annotated-cell-2-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">label =</span> <span class="fu">paste0</span>(COG_category, <span class="st">": "</span>, category_name))</span>
<span id="annotated-cell-2-13"><a href="#annotated-cell-2-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="annotated-cell-2-14"><a href="#annotated-cell-2-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Show first 10 lines of data frame</span></span>
<span id="annotated-cell-2-15"><a href="#annotated-cell-2-15" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(emapper_dt, <span class="dv">10</span>) <span class="sc">%&gt;%</span> <span class="fu">kbl</span>() <span class="sc">%&gt;%</span> <span class="fu">kable_styling</span>(<span class="at">full_width =</span> <span class="cn">FALSE</span>)</span>
<span id="annotated-cell-2-16"><a href="#annotated-cell-2-16" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="annotated-cell-2-17"><a href="#annotated-cell-2-17" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<button class="code-annotation-anchor" data-target-cell="annotated-cell-2" data-target-annotation="8">8</button><span id="annotated-cell-2-18" class="code-annotation-target"><a href="#annotated-cell-2-18" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(emapper_dt, <span class="fu">aes</span>(<span class="at">x =</span> COG_category, <span class="at">y =</span> n, <span class="at">fill =</span> label)) <span class="sc">+</span></span>
<button class="code-annotation-anchor" data-target-cell="annotated-cell-2" data-target-annotation="9">9</button><span id="annotated-cell-2-19" class="code-annotation-target"><a href="#annotated-cell-2-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>) <span class="sc">+</span></span>
<button class="code-annotation-anchor" data-target-cell="annotated-cell-2" data-target-annotation="10">10</button><span id="annotated-cell-2-20" class="code-annotation-target"><a href="#annotated-cell-2-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">100</span>, <span class="dv">10</span>)) <span class="sc">+</span></span>
<button class="code-annotation-anchor" data-target-cell="annotated-cell-2" data-target-annotation="11">11</button><span id="annotated-cell-2-21" class="code-annotation-target"><a href="#annotated-cell-2-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"COG category"</span>,</span>
<span id="annotated-cell-2-22"><a href="#annotated-cell-2-22" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"Number of sequences"</span>, </span>
<span id="annotated-cell-2-23"><a href="#annotated-cell-2-23" aria-hidden="true" tabindex="-1"></a>       <span class="at">fill =</span> <span class="st">"COG category"</span>) <span class="sc">+</span></span>
<button class="code-annotation-anchor" data-target-cell="annotated-cell-2" data-target-annotation="12">12</button><span id="annotated-cell-2-24" class="code-annotation-target"><a href="#annotated-cell-2-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.key.size =</span> <span class="fu">unit</span>(<span class="fl">0.5</span>, <span class="st">'cm'</span>))</span><div class="code-annotation-gutter-bg"></div><div class="code-annotation-gutter"></div></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-annotation">
<dl class="code-annotation-container-hidden code-annotation-container-grid">
<dt data-target-cell="annotated-cell-2" data-target-annotation="1">1</dt>
<dd>
<span data-code-annotation="1" data-code-lines="2" data-code-cell="annotated-cell-2"><code>cog_category.txt</code> is a cluster of orthologous groups (COG) dictionary. Use for look up the category names into the plot.</span>
</dd>
<dt data-target-cell="annotated-cell-2" data-target-annotation="2">2</dt>
<dd>
<span data-code-annotation="2" data-code-lines="5" data-code-cell="annotated-cell-2">Read eggNOG-mapper result from excel file into <code>emapper_dt</code>, using <code>read.xlsx()</code> from openxlsx library. The excel is read by skipping the first 3 rows and select only 7th column.</span>
</dd>
<dt data-target-cell="annotated-cell-2" data-target-annotation="3">3</dt>
<dd>
<span data-code-annotation="3" data-code-lines="8" data-code-cell="annotated-cell-2">Then, go to the next step by pipe <code>%&gt;%</code>. This step it to filter unclassified COGs <code>-</code> from the <code>COG_category</code> using dplyr <code>filter()</code>.</span>
</dd>
<dt data-target-cell="annotated-cell-2" data-target-annotation="4">4</dt>
<dd>
<span data-code-annotation="4" data-code-lines="9" data-code-cell="annotated-cell-2">Then group the data frame by <code>COG_category</code>.</span>
</dd>
<dt data-target-cell="annotated-cell-2" data-target-annotation="5">5</dt>
<dd>
<span data-code-annotation="5" data-code-lines="10" data-code-cell="annotated-cell-2">Count number of COGs presented in this eggNOG-mapper result.</span>
</dd>
<dt data-target-cell="annotated-cell-2" data-target-annotation="6">6</dt>
<dd>
<span data-code-annotation="6" data-code-lines="11" data-code-cell="annotated-cell-2">Merge COG dictionary into the result using dplyr <code>left_join()</code>. Two data frames are merged by matching the column <code>COG_category</code> from <code>emapper_dt</code> with the column <code>category</code> of <code>COG_dict</code>.</span>
</dd>
<dt data-target-cell="annotated-cell-2" data-target-annotation="7">7</dt>
<dd>
<span data-code-annotation="7" data-code-lines="12" data-code-cell="annotated-cell-2">Then, add new column <code>label</code> for the plot legend, by append COG category together with the category name.</span>
</dd>
<dt data-target-cell="annotated-cell-2" data-target-annotation="8">8</dt>
<dd>
<span data-code-annotation="8" data-code-lines="18" data-code-cell="annotated-cell-2">Plot the result from by showing <code>COG_category</code> in x-axis, number of COGs <code>n</code> in y-axis, fill and add legend by column <code>label</code>.</span>
</dd>
<dt data-target-cell="annotated-cell-2" data-target-annotation="9">9</dt>
<dd>
<span data-code-annotation="9" data-code-lines="19" data-code-cell="annotated-cell-2">Plot bar plot using <code>geom_bar()</code> and <code>stat = "identity"</code></span>
</dd>
<dt data-target-cell="annotated-cell-2" data-target-annotation="10">10</dt>
<dd>
<span data-code-annotation="10" data-code-lines="20" data-code-cell="annotated-cell-2">Set y-axis breaks</span>
</dd>
<dt data-target-cell="annotated-cell-2" data-target-annotation="11">11</dt>
<dd>
<span data-code-annotation="11" data-code-lines="21" data-code-cell="annotated-cell-2">Customize label of x- and y-axis, and legend name in <code>fill</code>.</span>
</dd>
<dt data-target-cell="annotated-cell-2" data-target-annotation="12">12</dt>
<dd>
<span data-code-annotation="12" data-code-lines="24" data-code-cell="annotated-cell-2">Adjust size of legend using <code>theme(legend.key.size)</code>.</span>
</dd>
</dl>
</div>
<div class="cell-output-display">
<p><img src="plotting_omics_data_files/figure-html/unnamed-chunk-2-1.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<table class="table table-sm table-striped small" data-quarto-postprocess="true">
<caption>First 10 lines of the number of COG category results. </caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th">COG_category</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">n</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">category_name</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">label</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">A</td>
<td style="text-align: right;">5</td>
<td style="text-align: left;">RNA processing and modification</td>
<td style="text-align: left;">A: RNA processing and modification</td>
</tr>
<tr class="even">
<td style="text-align: left;">B</td>
<td style="text-align: right;">1</td>
<td style="text-align: left;">Chromatin Structure and dynamics</td>
<td style="text-align: left;">B: Chromatin Structure and dynamics</td>
</tr>
<tr class="odd">
<td style="text-align: left;">C</td>
<td style="text-align: right;">5</td>
<td style="text-align: left;">Energy production and conversion</td>
<td style="text-align: left;">C: Energy production and conversion</td>
</tr>
<tr class="even">
<td style="text-align: left;">D</td>
<td style="text-align: right;">5</td>
<td style="text-align: left;">Cell cycle control and mitosis</td>
<td style="text-align: left;">D: Cell cycle control and mitosis</td>
</tr>
<tr class="odd">
<td style="text-align: left;">G</td>
<td style="text-align: right;">1</td>
<td style="text-align: left;">Carbohydrate metabolism and transport</td>
<td style="text-align: left;">G: Carbohydrate metabolism and transport</td>
</tr>
<tr class="even">
<td style="text-align: left;">I</td>
<td style="text-align: right;">4</td>
<td style="text-align: left;">Lipid metabolism</td>
<td style="text-align: left;">I: Lipid metabolism</td>
</tr>
<tr class="odd">
<td style="text-align: left;">J</td>
<td style="text-align: right;">6</td>
<td style="text-align: left;">Tranlsation</td>
<td style="text-align: left;">J: Tranlsation</td>
</tr>
<tr class="even">
<td style="text-align: left;">K</td>
<td style="text-align: right;">55</td>
<td style="text-align: left;">Transcription</td>
<td style="text-align: left;">K: Transcription</td>
</tr>
<tr class="odd">
<td style="text-align: left;">L</td>
<td style="text-align: right;">6</td>
<td style="text-align: left;">Replication and repair</td>
<td style="text-align: left;">L: Replication and repair</td>
</tr>
<tr class="even">
<td style="text-align: left;">M</td>
<td style="text-align: right;">4</td>
<td style="text-align: left;">Cell wall/membrane/envelop biogenesis</td>
<td style="text-align: left;">M: Cell wall/membrane/envelop biogenesis</td>
</tr>
</tbody>
</table>


</div>
</div>
</section>
<section id="principal-component-analysis" class="level2" data-number="3">
<h2 data-number="3" class="anchored" data-anchor-id="principal-component-analysis"><span class="header-section-number">3</span> Principal Component Analysis</h2>
<p>Typically, PCA is used for dimensionality reduction to get lower-dimensional data while keeping the most variation possible by taking only the first few principal components. In principle, SVD (singlular value decomposition) or eigenvalues of the data covariance matrix can be used to evaluate the principal components.</p>
<p>From <a href="https://www.datacamp.com/tutorial/pca-analysis-r">datacamp</a>, The PCA can be more easily computed by following these five steps:</p>
<ol type="1">
<li><p>Normalize data: As these data have different scales, performing PCA on them will result in a biased result. In order to ensure that each attribute contributes equally and to prevent one variable from dominating others, the data set needs to be normalized.</p></li>
<li><p>Computing the covariable matrix from the normalized data. This is a symmetric matrix, and each element (i, j) corresponds to the covariance between variables i and j.</p></li>
<li><p>Get Eigenvectors and eigenvalues: The term eigenvector refers to a direction in mathematics, such as “vertical” or “90 degrees”. By contrast, an eigenvalue represents how much variance there is in a given direction in the data. Therefore, each eigenvector has a corresponding eigenvalue.</p></li>
<li><p>Select the principal components: It’s about choosing the eigenvector with the highest eigenvalue that corresponds to the first principal component. The second principal component is the eigenvector with the second highest eigenvalue, etc.</p></li>
<li><p>Transforming data into a new form: A new subspace is defined by the principal components, so the original data is re-oriented by multiplying it by the previously computed eigenvectors. As a result of this transformation, the original data doesn’t change, but instead provides a new perspective to better represent it.</p></li>
</ol>
<p>We will demonstrate PCA on a <em>pasilla</em> data set <span class="citation" data-cites="brooks2010">(<a href="#ref-brooks2010" role="doc-biblioref">Brooks et al. 2010</a>)</span>. This data set was obtained from an experiment on <em>Drosophila melanogaster</em> cell cultures that investigated the effects of knocking down the splicing factor <em>pasilla</em> using RNAi.</p>
<div class="cell" data-tbl-subcap="[&quot;First 10 rows of de_input&quot;,&quot;Example of coldata&quot;]">
<div class="sourceCode cell-code" id="annotated-cell-3"><pre class="sourceCode r code-annotation-code code-with-copy code-annotated"><code class="sourceCode r"><span id="annotated-cell-3-1"><a href="#annotated-cell-3-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load data set</span></span>
<span id="annotated-cell-3-2"><a href="#annotated-cell-3-2" aria-hidden="true" tabindex="-1"></a>cts <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(<span class="st">"https://raw.githubusercontent.com/JirathNuan/r-handviz-workshop/main/datasets/cts.tsv"</span>)</span>
<span id="annotated-cell-3-3"><a href="#annotated-cell-3-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="annotated-cell-3-4"><a href="#annotated-cell-3-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Prepare DESeq input, which is expecting a matrix of integers.</span></span>
<button class="code-annotation-anchor" data-target-cell="annotated-cell-3" data-target-annotation="1">1</button><span id="annotated-cell-3-5" class="code-annotation-target"><a href="#annotated-cell-3-5" aria-hidden="true" tabindex="-1"></a>de_input <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(cts[,<span class="sc">-</span><span class="dv">1</span>])</span>
<span id="annotated-cell-3-6"><a href="#annotated-cell-3-6" aria-hidden="true" tabindex="-1"></a><span class="fu">row.names</span>(de_input)<span class="ot">&lt;-</span> cts<span class="sc">$</span>transcript_name</span>
<span id="annotated-cell-3-7"><a href="#annotated-cell-3-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove NAs</span></span>
<button class="code-annotation-anchor" data-target-cell="annotated-cell-3" data-target-annotation="2">2</button><span id="annotated-cell-3-8" class="code-annotation-target"><a href="#annotated-cell-3-8" aria-hidden="true" tabindex="-1"></a>de_input <span class="ot">&lt;-</span> de_input[<span class="fu">complete.cases</span>(de_input), ]</span>
<span id="annotated-cell-3-9"><a href="#annotated-cell-3-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="annotated-cell-3-10"><a href="#annotated-cell-3-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Show first 10 rows of the matrix</span></span>
<span id="annotated-cell-3-11"><a href="#annotated-cell-3-11" aria-hidden="true" tabindex="-1"></a><span class="fu">kbl</span>(<span class="fu">head</span>(de_input, <span class="dv">10</span>))</span>
<span id="annotated-cell-3-12"><a href="#annotated-cell-3-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="annotated-cell-3-13"><a href="#annotated-cell-3-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Create sample metadata     </span></span>
<button class="code-annotation-anchor" data-target-cell="annotated-cell-3" data-target-annotation="3">3</button><span id="annotated-cell-3-14" class="code-annotation-target"><a href="#annotated-cell-3-14" aria-hidden="true" tabindex="-1"></a>coldata <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">sample =</span> <span class="fu">colnames</span>(de_input),</span>
<span id="annotated-cell-3-15"><a href="#annotated-cell-3-15" aria-hidden="true" tabindex="-1"></a>                      <span class="at">sample_group =</span> <span class="fu">gsub</span>(<span class="st">"[0-9]"</span>, <span class="st">""</span>, <span class="fu">colnames</span>(de_input)))</span>
<span id="annotated-cell-3-16"><a href="#annotated-cell-3-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="annotated-cell-3-17"><a href="#annotated-cell-3-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="annotated-cell-3-18"><a href="#annotated-cell-3-18" aria-hidden="true" tabindex="-1"></a><span class="co"># Show how experimental data looks like</span></span>
<span id="annotated-cell-3-19"><a href="#annotated-cell-3-19" aria-hidden="true" tabindex="-1"></a><span class="fu">kbl</span>(coldata)</span><div class="code-annotation-gutter-bg"></div><div class="code-annotation-gutter"></div></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-annotation">
<dl class="code-annotation-container-hidden code-annotation-container-grid">
<dt data-target-cell="annotated-cell-3" data-target-annotation="1">1</dt>
<dd>
<span data-code-annotation="1" data-code-lines="5" data-code-cell="annotated-cell-3">Load data set into <code>cts</code> data frame, and convert to matrix.</span>
</dd>
<dt data-target-cell="annotated-cell-3" data-target-annotation="2">2</dt>
<dd>
<span data-code-annotation="2" data-code-lines="8" data-code-cell="annotated-cell-3">Removing NA. The whole row will be deleted if NA is observed.</span>
</dd>
<dt data-target-cell="annotated-cell-3" data-target-annotation="3">3</dt>
<dd>
<span data-code-annotation="3" data-code-lines="14" data-code-cell="annotated-cell-3">Sample metadata should show a relationship between the sample name (from the matrix column), the sample group, and other experimental design.</span>
</dd>
</dl>
</div>
<div class="cell-output-display">
<table data-quarto-postprocess="true" class="table table-sm table-striped small">
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th"></th>
<th style="text-align: right;" data-quarto-table-cell-role="th">treated1</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">treated2</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">treated3</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">untreated1</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">untreated2</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">untreated3</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">untreated4</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">FBgn0000003</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
</tr>
<tr class="even">
<td style="text-align: left;">FBgn0000008</td>
<td style="text-align: right;">140</td>
<td style="text-align: right;">88</td>
<td style="text-align: right;">70</td>
<td style="text-align: right;">92</td>
<td style="text-align: right;">161</td>
<td style="text-align: right;">76</td>
<td style="text-align: right;">70</td>
</tr>
<tr class="odd">
<td style="text-align: left;">FBgn0000014</td>
<td style="text-align: right;">4</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">5</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
</tr>
<tr class="even">
<td style="text-align: left;">FBgn0000015</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">2</td>
</tr>
<tr class="odd">
<td style="text-align: left;">FBgn0000017</td>
<td style="text-align: right;">6205</td>
<td style="text-align: right;">3072</td>
<td style="text-align: right;">3334</td>
<td style="text-align: right;">4664</td>
<td style="text-align: right;">8714</td>
<td style="text-align: right;">3564</td>
<td style="text-align: right;">3150</td>
</tr>
<tr class="even">
<td style="text-align: left;">FBgn0000018</td>
<td style="text-align: right;">722</td>
<td style="text-align: right;">299</td>
<td style="text-align: right;">308</td>
<td style="text-align: right;">583</td>
<td style="text-align: right;">761</td>
<td style="text-align: right;">245</td>
<td style="text-align: right;">310</td>
</tr>
<tr class="odd">
<td style="text-align: left;">FBgn0000022</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
</tr>
<tr class="even">
<td style="text-align: left;">FBgn0000024</td>
<td style="text-align: right;">10</td>
<td style="text-align: right;">7</td>
<td style="text-align: right;">5</td>
<td style="text-align: right;">10</td>
<td style="text-align: right;">11</td>
<td style="text-align: right;">3</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="odd">
<td style="text-align: left;">FBgn0000028</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
</tr>
<tr class="even">
<td style="text-align: left;">FBgn0000032</td>
<td style="text-align: right;">1698</td>
<td style="text-align: right;">696</td>
<td style="text-align: right;">757</td>
<td style="text-align: right;">1446</td>
<td style="text-align: right;">1713</td>
<td style="text-align: right;">615</td>
<td style="text-align: right;">672</td>
</tr>
</tbody>
</table>
<table data-quarto-postprocess="true" class="table table-sm table-striped small">
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th">sample</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">sample_group</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">treated1</td>
<td style="text-align: left;">treated</td>
</tr>
<tr class="even">
<td style="text-align: left;">treated2</td>
<td style="text-align: left;">treated</td>
</tr>
<tr class="odd">
<td style="text-align: left;">treated3</td>
<td style="text-align: left;">treated</td>
</tr>
<tr class="even">
<td style="text-align: left;">untreated1</td>
<td style="text-align: left;">untreated</td>
</tr>
<tr class="odd">
<td style="text-align: left;">untreated2</td>
<td style="text-align: left;">untreated</td>
</tr>
<tr class="even">
<td style="text-align: left;">untreated3</td>
<td style="text-align: left;">untreated</td>
</tr>
<tr class="odd">
<td style="text-align: left;">untreated4</td>
<td style="text-align: left;">untreated</td>
</tr>
</tbody>
</table>


</div>
</div>
<p>This workshop will demonstrate data normalization using <code>DESeq2</code> library.</p>
<div class="cell">
<div class="sourceCode cell-code" id="annotated-cell-4"><pre class="sourceCode r code-annotation-code code-with-copy code-annotated"><code class="sourceCode r"><span id="annotated-cell-4-1"><a href="#annotated-cell-4-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create DESeq object by load matrix and experimental design</span></span>
<button class="code-annotation-anchor" data-target-cell="annotated-cell-4" data-target-annotation="1">1</button><span id="annotated-cell-4-2" class="code-annotation-target"><a href="#annotated-cell-4-2" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> de_input,</span>
<span id="annotated-cell-4-3"><a href="#annotated-cell-4-3" aria-hidden="true" tabindex="-1"></a>                              <span class="at">colData =</span> coldata,</span>
<span id="annotated-cell-4-4"><a href="#annotated-cell-4-4" aria-hidden="true" tabindex="-1"></a>                              <span class="at">design=</span> <span class="sc">~</span> sample_group)</span>
<span id="annotated-cell-4-5"><a href="#annotated-cell-4-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="annotated-cell-4-6"><a href="#annotated-cell-4-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Perform differential expression analysis</span></span>
<button class="code-annotation-anchor" data-target-cell="annotated-cell-4" data-target-annotation="2">2</button><span id="annotated-cell-4-7" class="code-annotation-target"><a href="#annotated-cell-4-7" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds)</span>
<span id="annotated-cell-4-8"><a href="#annotated-cell-4-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="annotated-cell-4-9"><a href="#annotated-cell-4-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a normalized matrix of cts data set</span></span>
<button class="code-annotation-anchor" data-target-cell="annotated-cell-4" data-target-annotation="3">3</button><span id="annotated-cell-4-10" class="code-annotation-target"><a href="#annotated-cell-4-10" aria-hidden="true" tabindex="-1"></a>cts_norm <span class="ot">&lt;-</span> <span class="fu">counts</span>(dds, <span class="at">normalized =</span> <span class="cn">TRUE</span>)</span>
<span id="annotated-cell-4-11"><a href="#annotated-cell-4-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="annotated-cell-4-12"><a href="#annotated-cell-4-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Show how the data looks like</span></span>
<span id="annotated-cell-4-13"><a href="#annotated-cell-4-13" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(cts_norm) <span class="sc">%&gt;%</span> <span class="fu">kbl</span>()</span><div class="code-annotation-gutter-bg"></div><div class="code-annotation-gutter"></div></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-annotation">
<dl class="code-annotation-container-hidden code-annotation-container-grid">
<dt data-target-cell="annotated-cell-4" data-target-annotation="1">1</dt>
<dd>
<span data-code-annotation="1" data-code-lines="2" data-code-cell="annotated-cell-4">Create DESeq object using count matrix and sample metadata.</span>
</dd>
<dt data-target-cell="annotated-cell-4" data-target-annotation="2">2</dt>
<dd>
<span data-code-annotation="2" data-code-lines="7" data-code-cell="annotated-cell-4">This function is from DESeq2, to perform differential expression analysis based on the Negative Binomial (a.k.a. Gamma-Poisson) distribution. The analysis started with estimating size factors, estimating dispersion, and fitting the Negative Binomial GLM model and calculate Wald statistics.</span>
</dd>
<dt data-target-cell="annotated-cell-4" data-target-annotation="3">3</dt>
<dd>
<span data-code-annotation="3" data-code-lines="10" data-code-cell="annotated-cell-4">Create a matrix of normalized count from DESeq2 object after DE analysis.</span>
</dd>
</dl>
</div>
<div class="cell-output-display">
<table data-quarto-postprocess="true" class="table table-sm table-striped small">
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th"></th>
<th style="text-align: right;" data-quarto-table-cell-role="th">treated1</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">treated2</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">treated3</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">untreated1</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">untreated2</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">untreated3</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">untreated4</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">FBgn0000003</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">0.0000</td>
<td style="text-align: right;">1.200981</td>
<td style="text-align: right;">0.000000</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">0.000000</td>
<td style="text-align: right;">0.000000</td>
</tr>
<tr class="even">
<td style="text-align: left;">FBgn0000008</td>
<td style="text-align: right;">85.5968034</td>
<td style="text-align: right;">115.5963</td>
<td style="text-align: right;">84.068670</td>
<td style="text-align: right;">80.824908</td>
<td style="text-align: right;">89.7936241</td>
<td style="text-align: right;">117.004615</td>
<td style="text-align: right;">93.123591</td>
</tr>
<tr class="odd">
<td style="text-align: left;">FBgn0000014</td>
<td style="text-align: right;">2.4456230</td>
<td style="text-align: right;">0.0000</td>
<td style="text-align: right;">0.000000</td>
<td style="text-align: right;">4.392658</td>
<td style="text-align: right;">0.5577244</td>
<td style="text-align: right;">0.000000</td>
<td style="text-align: right;">0.000000</td>
</tr>
<tr class="even">
<td style="text-align: left;">FBgn0000015</td>
<td style="text-align: right;">0.6114057</td>
<td style="text-align: right;">0.0000</td>
<td style="text-align: right;">0.000000</td>
<td style="text-align: right;">0.000000</td>
<td style="text-align: right;">1.1154487</td>
<td style="text-align: right;">1.539534</td>
<td style="text-align: right;">2.660674</td>
</tr>
<tr class="odd">
<td style="text-align: left;">FBgn0000017</td>
<td style="text-align: right;">3793.7726082</td>
<td style="text-align: right;">4035.3632</td>
<td style="text-align: right;">4004.070676</td>
<td style="text-align: right;">4097.471407</td>
<td style="text-align: right;">4860.0101913</td>
<td style="text-align: right;">5486.900612</td>
<td style="text-align: right;">4190.561607</td>
</tr>
<tr class="even">
<td style="text-align: left;">FBgn0000018</td>
<td style="text-align: right;">441.4349433</td>
<td style="text-align: right;">392.7648</td>
<td style="text-align: right;">369.902150</td>
<td style="text-align: right;">512.183926</td>
<td style="text-align: right;">424.4282483</td>
<td style="text-align: right;">377.185929</td>
<td style="text-align: right;">412.404476</td>
</tr>
</tbody>
</table>


</div>
</div>
<p>This step is to transform the count matrix through Rlog (regularized log) transforms. Which transform the original count data into log2 scale by fitting a model with a term for each sample and a prior distribution on the coefficients. Use before plotting PCA.</p>
<div class="cell">
<div class="sourceCode cell-code" id="annotated-cell-5"><pre class="sourceCode r code-annotation-code code-with-copy code-annotated"><code class="sourceCode r"><span id="annotated-cell-5-1"><a href="#annotated-cell-5-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot PCA With pcaExplorer</span></span>
<button class="code-annotation-anchor" data-target-cell="annotated-cell-5" data-target-annotation="1">1</button><span id="annotated-cell-5-2" class="code-annotation-target"><a href="#annotated-cell-5-2" aria-hidden="true" tabindex="-1"></a>rld_dds <span class="ot">&lt;-</span> <span class="fu">rlogTransformation</span>(dds)</span>
<button class="code-annotation-anchor" data-target-cell="annotated-cell-5" data-target-annotation="2">2</button><span id="annotated-cell-5-3" class="code-annotation-target"><a href="#annotated-cell-5-3" aria-hidden="true" tabindex="-1"></a>pca_dds <span class="ot">&lt;-</span> <span class="fu">pcaplot</span>(</span>
<span id="annotated-cell-5-4"><a href="#annotated-cell-5-4" aria-hidden="true" tabindex="-1"></a>  rld_dds,</span>
<span id="annotated-cell-5-5"><a href="#annotated-cell-5-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">intgroup =</span> <span class="st">"sample_group"</span>,</span>
<span id="annotated-cell-5-6"><a href="#annotated-cell-5-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">ntop =</span> <span class="cn">Inf</span>,</span>
<span id="annotated-cell-5-7"><a href="#annotated-cell-5-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">pcX =</span> <span class="dv">1</span>,</span>
<span id="annotated-cell-5-8"><a href="#annotated-cell-5-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">pcY =</span> <span class="dv">2</span>,</span>
<span id="annotated-cell-5-9"><a href="#annotated-cell-5-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">title =</span> <span class="st">""</span>,</span>
<span id="annotated-cell-5-10"><a href="#annotated-cell-5-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">text_labels =</span> <span class="cn">TRUE</span></span>
<span id="annotated-cell-5-11"><a href="#annotated-cell-5-11" aria-hidden="true" tabindex="-1"></a>) <span class="sc">+</span></span>
<span id="annotated-cell-5-12"><a href="#annotated-cell-5-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_npg</span>() <span class="sc">+</span></span>
<span id="annotated-cell-5-13"><a href="#annotated-cell-5-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_npg</span>() <span class="sc">+</span></span>
<span id="annotated-cell-5-14"><a href="#annotated-cell-5-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"bottom"</span>) <span class="sc">+</span></span>
<span id="annotated-cell-5-15"><a href="#annotated-cell-5-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>()</span>
<span id="annotated-cell-5-16"><a href="#annotated-cell-5-16" aria-hidden="true" tabindex="-1"></a>pca_dds</span><div class="code-annotation-gutter-bg"></div><div class="code-annotation-gutter"></div></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-annotation">
<dl class="code-annotation-container-hidden code-annotation-container-grid">
<dt data-target-cell="annotated-cell-5" data-target-annotation="1">1</dt>
<dd>
<span data-code-annotation="1" data-code-lines="2" data-code-cell="annotated-cell-5">Rlog (regularized log) transforms the original count data into log2 scale by fitting a model with a term for each sample and a prior distribution on the coefficients. Use before plotting PCA.</span>
</dd>
<dt data-target-cell="annotated-cell-5" data-target-annotation="2">2</dt>
<dd>
<span data-code-annotation="2" data-code-lines="3" data-code-cell="annotated-cell-5">Plot PCA</span>
</dd>
</dl>
</div>
<div class="cell-output-display">
<p><img src="plotting_omics_data_files/figure-html/unnamed-chunk-5-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>This step is to examine the principal components and visualizing eigenvalues as a scree plot. A scree plot is used to determine the number of significant principal components in a set of data. It is a line graph that shows the eigenvalues of each component in descending order. This helps to identify which components are most important and which can be discarded.</p>
<p>We’ll demonstrate a scree plot using <code>fviz_eig()</code> function from <code>factoextra</code> library.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Scree plot</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>pcaobj_dds <span class="ot">&lt;-</span> <span class="fu">prcomp</span>(<span class="fu">t</span>(<span class="fu">assay</span>(rld_dds)))</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualizing scree plot</span></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a><span class="fu">fviz_eig</span>(pcaobj_dds, </span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>         <span class="at">addlabels =</span> <span class="cn">TRUE</span>, </span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>         <span class="at">barfill =</span> <span class="fu">pal_npg</span>()(<span class="dv">1</span>),</span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a>         <span class="at">barcolor =</span> <span class="st">"black"</span>,</span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a>         <span class="at">title =</span> <span class="st">"Proportion of explained proportion of variance - cts data set"</span>,</span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a>         <span class="at">ggtheme =</span> <span class="fu">theme_gray</span>())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="plotting_omics_data_files/figure-html/unnamed-chunk-6-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>Then, extract genes that are mostly contributed in the principal components. By extracting genes that are primarily responsible for the principal components, it becomes easier to understand the underlying structure of the data and identify potential observations that are associated with certain variables.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co"># extract the table of the genes with high loadings</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>top100_pc <span class="ot">&lt;-</span> <span class="fu">hi_loadings</span>(pcaobj_dds,</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>                         <span class="at">topN =</span> <span class="dv">100</span>,</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>                         <span class="at">exprTable =</span> <span class="fu">counts</span>(dds))</span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Show how experimental data looks like</span></span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(top100_pc) <span class="sc">%&gt;%</span> <span class="fu">kbl</span>() <span class="sc">%&gt;%</span> <span class="fu">kable_styling</span>(<span class="at">full_width =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<table class="table table-sm table-striped small" data-quarto-postprocess="true">
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th"></th>
<th style="text-align: right;" data-quarto-table-cell-role="th">treated1</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">treated2</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">treated3</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">untreated1</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">untreated2</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">untreated3</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">untreated4</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">FBgn0044047</td>
<td style="text-align: right;">348</td>
<td style="text-align: right;">118</td>
<td style="text-align: right;">151</td>
<td style="text-align: right;">428</td>
<td style="text-align: right;">667</td>
<td style="text-align: right;">200</td>
<td style="text-align: right;">235</td>
</tr>
<tr class="even">
<td style="text-align: left;">FBgn0037683</td>
<td style="text-align: right;">1496</td>
<td style="text-align: right;">710</td>
<td style="text-align: right;">918</td>
<td style="text-align: right;">1868</td>
<td style="text-align: right;">2848</td>
<td style="text-align: right;">1090</td>
<td style="text-align: right;">1641</td>
</tr>
<tr class="odd">
<td style="text-align: left;">FBgn0038381</td>
<td style="text-align: right;">150</td>
<td style="text-align: right;">74</td>
<td style="text-align: right;">79</td>
<td style="text-align: right;">260</td>
<td style="text-align: right;">309</td>
<td style="text-align: right;">121</td>
<td style="text-align: right;">178</td>
</tr>
<tr class="even">
<td style="text-align: left;">FBgn0040752</td>
<td style="text-align: right;">5117</td>
<td style="text-align: right;">2778</td>
<td style="text-align: right;">2786</td>
<td style="text-align: right;">5293</td>
<td style="text-align: right;">10550</td>
<td style="text-align: right;">4998</td>
<td style="text-align: right;">4235</td>
</tr>
<tr class="odd">
<td style="text-align: left;">FBgn0029801</td>
<td style="text-align: right;">1628</td>
<td style="text-align: right;">788</td>
<td style="text-align: right;">917</td>
<td style="text-align: right;">1802</td>
<td style="text-align: right;">3677</td>
<td style="text-align: right;">1256</td>
<td style="text-align: right;">1313</td>
</tr>
<tr class="even">
<td style="text-align: left;">FBgn0037635</td>
<td style="text-align: right;">167</td>
<td style="text-align: right;">71</td>
<td style="text-align: right;">92</td>
<td style="text-align: right;">229</td>
<td style="text-align: right;">378</td>
<td style="text-align: right;">139</td>
<td style="text-align: right;">181</td>
</tr>
</tbody>
</table>


</div>
</div>
<p>Then, plot PCA biplot<code>genespca()</code> by compute the principal components of the genes, eventually displaying the samples as in a typical biplot visualization.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>groups_cts <span class="ot">&lt;-</span> <span class="fu">colData</span>(dds)<span class="sc">$</span>sample_group</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>cols_cts <span class="ot">&lt;-</span> scales<span class="sc">::</span><span class="fu">hue_pal</span>()(<span class="dv">2</span>)[groups_cts]</span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a><span class="co"># with many genes, do not plot the labels of the genes</span></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a><span class="fu">genespca</span>(</span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a>  rld_dds,</span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">ntop =</span> <span class="dv">100</span>,</span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">choices =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>),</span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">arrowColors =</span> cols_cts,</span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">groupNames =</span> groups_cts,</span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a>  <span class="at">useRownamesAsLabels =</span> <span class="cn">FALSE</span>,</span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a>  <span class="at">varname.size =</span> <span class="dv">5</span>,</span>
<span id="cb4-14"><a href="#cb4-14" aria-hidden="true" tabindex="-1"></a>  <span class="at">biplot =</span> <span class="cn">TRUE</span>,</span>
<span id="cb4-15"><a href="#cb4-15" aria-hidden="true" tabindex="-1"></a>  <span class="at">alpha =</span> <span class="fl">0.5</span>,</span>
<span id="cb4-16"><a href="#cb4-16" aria-hidden="true" tabindex="-1"></a>  <span class="at">point_size =</span> <span class="fl">2.5</span></span>
<span id="cb4-17"><a href="#cb4-17" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="plotting_omics_data_files/figure-html/unnamed-chunk-8-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>Plots the distribution of expression values, either with density lines, boxplots or violin plots.</p>
<div>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">distro_expr</span>(rld_dds, <span class="at">plot_type =</span> <span class="st">"violin"</span>) <span class="sc">+</span> </span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_npg</span>() <span class="sc">+</span> </span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_npg</span>()</span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a><span class="fu">distro_expr</span>(rld_dds, <span class="at">plot_type =</span> <span class="st">"boxplot"</span>) <span class="sc">+</span> </span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_npg</span>() <span class="sc">+</span> </span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_npg</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell quarto-layout-panel">
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="cell-output-display quarto-layout-cell" style="flex-basis: 33.3%;justify-content: center;">
<p><img src="plotting_omics_data_files/figure-html/unnamed-chunk-9-1.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display quarto-layout-cell" style="flex-basis: 33.3%;justify-content: center;">
<p><img src="plotting_omics_data_files/figure-html/unnamed-chunk-9-2.png" class="img-fluid" width="672"></p>
</div>
</div>
</div>
</div>
<blockquote class="blockquote">
<p>See more usage of <strong>pcaExplorer</strong>: <a href="https://federicomarini.github.io/pcaExplorer/articles/pcaExplorer.html#functions-exported-by-the-package-for-standalone-usage" class="uri">https://federicomarini.github.io/pcaExplorer/articles/pcaExplorer.html#functions-exported-by-the-package-for-standalone-usage</a></p>
</blockquote>
</section>
<section id="hierarchical-clustering-analysis-heat-maps" class="level2" data-number="4">
<h2 data-number="4" class="anchored" data-anchor-id="hierarchical-clustering-analysis-heat-maps"><span class="header-section-number">4</span> Hierarchical Clustering Analysis (Heat maps)</h2>
<p>Hierarchical clustering is an unsupervised machine learning algorithm. It is used to group data points into clusters based on their similarity. The algorithm gradually merges or splits clusters after creating a hierarchy. As a result, a dendrogram shows which points are in each cluster and how similar they are.</p>
<p>We’ll demonstrate the hierarchical clustering and plot heat map using <code>pheatmap()</code>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create annotation column</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>annot_column <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">sample =</span> <span class="fu">colnames</span>(top100_pc),</span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>                           <span class="at">group =</span> <span class="fu">gsub</span>(<span class="st">"[0-9]"</span>, <span class="st">""</span>, <span class="fu">colnames</span>(top100_pc))) <span class="sc">%&gt;%</span> </span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(<span class="at">var =</span> <span class="st">"sample"</span>)</span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>annot_column <span class="sc">%&gt;%</span> <span class="fu">kbl</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<table data-quarto-postprocess="true" class="table table-sm table-striped small">
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th"></th>
<th style="text-align: left;" data-quarto-table-cell-role="th">group</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">treated1</td>
<td style="text-align: left;">treated</td>
</tr>
<tr class="even">
<td style="text-align: left;">treated2</td>
<td style="text-align: left;">treated</td>
</tr>
<tr class="odd">
<td style="text-align: left;">treated3</td>
<td style="text-align: left;">treated</td>
</tr>
<tr class="even">
<td style="text-align: left;">untreated1</td>
<td style="text-align: left;">untreated</td>
</tr>
<tr class="odd">
<td style="text-align: left;">untreated2</td>
<td style="text-align: left;">untreated</td>
</tr>
<tr class="even">
<td style="text-align: left;">untreated3</td>
<td style="text-align: left;">untreated</td>
</tr>
<tr class="odd">
<td style="text-align: left;">untreated4</td>
<td style="text-align: left;">untreated</td>
</tr>
</tbody>
</table>


</div>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a list of annotation color</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>sample_pal <span class="ot">&lt;-</span> <span class="fu">pal_npg</span>()(<span class="dv">2</span>)</span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>annot_colors <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">group =</span> <span class="fu">c</span>(<span class="at">treated =</span> sample_pal[<span class="dv">1</span>],</span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>                               <span class="at">untreated =</span> sample_pal[<span class="dv">2</span>]))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ul>
<li>Heat map of Top 100 genes that are mostly contributed in the principal components above.</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Heatmap of the top 100 principal components</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>hm_top100pc <span class="ot">&lt;-</span> <span class="fu">pheatmap</span>(</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>  top100_pc,</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">scale =</span> <span class="st">"row"</span>,</span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">color =</span> <span class="fu">colorRampPalette</span>(<span class="fu">rev</span>(<span class="fu">brewer.pal</span>(<span class="at">n =</span> <span class="dv">7</span>, <span class="at">name =</span> <span class="st">"RdBu"</span>)))(<span class="dv">100</span>),</span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">show_rownames =</span> <span class="cn">FALSE</span>,</span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">name =</span> <span class="st">"Normalized count"</span>,</span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">clustering_distance_rows =</span> <span class="st">"euclidean"</span>,</span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">clustering_distance_cols =</span> <span class="st">"euclidean"</span>,</span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">clustering_method =</span> <span class="st">"complete"</span>,</span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">annotation_col =</span> annot_column,</span>
<span id="cb8-12"><a href="#cb8-12" aria-hidden="true" tabindex="-1"></a>  <span class="at">annotation_names_col =</span> <span class="cn">FALSE</span>,</span>
<span id="cb8-13"><a href="#cb8-13" aria-hidden="true" tabindex="-1"></a>  <span class="at">annotation_colors =</span> annot_colors,</span>
<span id="cb8-14"><a href="#cb8-14" aria-hidden="true" tabindex="-1"></a>  <span class="at">cutree_rows =</span> <span class="dv">3</span>,</span>
<span id="cb8-15"><a href="#cb8-15" aria-hidden="true" tabindex="-1"></a>  <span class="at">annotation_legend =</span> <span class="cn">FALSE</span></span>
<span id="cb8-16"><a href="#cb8-16" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb8-17"><a href="#cb8-17" aria-hidden="true" tabindex="-1"></a>hm_top100pc</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="plotting_omics_data_files/figure-html/unnamed-chunk-11-1.png" class="img-fluid" width="480"></p>
</div>
</div>
<ul>
<li>Heat map of differentially expressed genes.</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Identify differentially expressed genes</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>dds_result <span class="ot">&lt;-</span> <span class="fu">results</span>(<span class="at">object =</span> dds,</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>                      <span class="at">contrast =</span> <span class="fu">c</span>(<span class="st">"sample_group"</span>, <span class="st">"untreated"</span>, <span class="st">"treated"</span>),</span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>                      <span class="at">tidy =</span> <span class="cn">TRUE</span>,</span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a>                      <span class="at">pAdjustMethod =</span> <span class="st">"fdr"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Filter DEG, we’ll use absolute log2FC &gt; 1 and FDR &lt; 0.05</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>dds_result_deg <span class="ot">&lt;-</span> dds_result <span class="sc">%&gt;%</span> </span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(padj <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> <span class="fu">abs</span>(log2FoldChange) <span class="sc">&gt;</span> <span class="dv">1</span>)</span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract count matrix of DEGenes</span></span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>deg_norm_count <span class="ot">&lt;-</span> cts_norm[<span class="fu">rownames</span>(cts_norm) <span class="sc">%in%</span> dds_result_deg<span class="sc">$</span>row, ]</span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot heatmap of DEGs</span></span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Heatmap of the top 100 principal components</span></span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true" tabindex="-1"></a>hm_deg <span class="ot">&lt;-</span> <span class="fu">pheatmap</span>(</span>
<span id="cb10-10"><a href="#cb10-10" aria-hidden="true" tabindex="-1"></a>  deg_norm_count,</span>
<span id="cb10-11"><a href="#cb10-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">scale =</span> <span class="st">"row"</span>,</span>
<span id="cb10-12"><a href="#cb10-12" aria-hidden="true" tabindex="-1"></a>  <span class="at">color =</span> <span class="fu">colorRampPalette</span>(<span class="fu">rev</span>(<span class="fu">brewer.pal</span>(<span class="at">n =</span> <span class="dv">7</span>, <span class="at">name =</span> <span class="st">"RdBu"</span>)))(<span class="dv">100</span>),</span>
<span id="cb10-13"><a href="#cb10-13" aria-hidden="true" tabindex="-1"></a>  <span class="at">show_rownames =</span> <span class="cn">FALSE</span>,</span>
<span id="cb10-14"><a href="#cb10-14" aria-hidden="true" tabindex="-1"></a>  <span class="at">name =</span> <span class="st">"Normalized count"</span>,</span>
<span id="cb10-15"><a href="#cb10-15" aria-hidden="true" tabindex="-1"></a>  <span class="at">clustering_distance_rows =</span> <span class="st">"euclidean"</span>,</span>
<span id="cb10-16"><a href="#cb10-16" aria-hidden="true" tabindex="-1"></a>  <span class="at">clustering_distance_cols =</span> <span class="st">"euclidean"</span>,</span>
<span id="cb10-17"><a href="#cb10-17" aria-hidden="true" tabindex="-1"></a>  <span class="at">clustering_method =</span> <span class="st">"complete"</span>,</span>
<span id="cb10-18"><a href="#cb10-18" aria-hidden="true" tabindex="-1"></a>  <span class="at">annotation_col =</span> annot_column,</span>
<span id="cb10-19"><a href="#cb10-19" aria-hidden="true" tabindex="-1"></a>  <span class="at">annotation_names_col =</span> <span class="cn">FALSE</span>,</span>
<span id="cb10-20"><a href="#cb10-20" aria-hidden="true" tabindex="-1"></a>  <span class="at">annotation_colors =</span> annot_colors,</span>
<span id="cb10-21"><a href="#cb10-21" aria-hidden="true" tabindex="-1"></a>  <span class="at">cutree_rows =</span> <span class="dv">2</span>,</span>
<span id="cb10-22"><a href="#cb10-22" aria-hidden="true" tabindex="-1"></a>  <span class="at">annotation_legend =</span> <span class="cn">FALSE</span>)</span>
<span id="cb10-23"><a href="#cb10-23" aria-hidden="true" tabindex="-1"></a>hm_deg</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="plotting_omics_data_files/figure-html/unnamed-chunk-13-1.png" class="img-fluid" width="480"></p>
</div>
</div>
<blockquote class="blockquote">
<p>See more <code>pheatmap()</code> usage: <a href="https://www.reneshbedre.com/blog/heatmap-with-pheatmap-package-r.html" class="uri">https://www.reneshbedre.com/blog/heatmap-with-pheatmap-package-r.html</a></p>
</blockquote>
</section>
<section id="volcano-plot" class="level2" data-number="5">
<h2 data-number="5" class="anchored" data-anchor-id="volcano-plot"><span class="header-section-number">5</span> Volcano Plot</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>volcano_deg <span class="ot">&lt;-</span> dds_result <span class="sc">%&gt;%</span> </span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Expression =</span> <span class="fu">case_when</span>(</span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>      log2FoldChange <span class="sc">&gt;=</span> <span class="dv">1</span> <span class="sc">&amp;</span> padj <span class="sc">&lt;=</span> <span class="fl">0.05</span> <span class="sc">~</span> <span class="st">"Upregulated"</span>,</span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>      log2FoldChange <span class="sc">&lt;=</span> <span class="sc">-</span><span class="dv">1</span> <span class="sc">&amp;</span> padj <span class="sc">&lt;=</span> <span class="fl">0.05</span> <span class="sc">~</span> <span class="st">"Downregulated"</span>,</span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>      <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"Unchanged"</span>))</span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a>p_volcano <span class="ot">&lt;-</span> <span class="fu">EnhancedVolcano</span>(</span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a>  volcano_deg,</span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">lab =</span> volcano_deg<span class="sc">$</span>row,</span>
<span id="cb11-10"><a href="#cb11-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">x =</span> <span class="st">'log2FoldChange'</span>,</span>
<span id="cb11-11"><a href="#cb11-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">y =</span> <span class="st">'padj'</span>,</span>
<span id="cb11-12"><a href="#cb11-12" aria-hidden="true" tabindex="-1"></a>  <span class="at">xlim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">5</span>,<span class="dv">5</span>),</span>
<span id="cb11-13"><a href="#cb11-13" aria-hidden="true" tabindex="-1"></a>  <span class="at">ylab =</span> <span class="fu">bquote</span>( <span class="sc">~</span> <span class="sc">-</span>Log[<span class="dv">10</span>] <span class="sc">~</span> <span class="fu">italic</span>(FDR)),</span>
<span id="cb11-14"><a href="#cb11-14" aria-hidden="true" tabindex="-1"></a>  <span class="at">title =</span> <span class="cn">NULL</span>,</span>
<span id="cb11-15"><a href="#cb11-15" aria-hidden="true" tabindex="-1"></a>  <span class="at">caption =</span> <span class="cn">NULL</span>,</span>
<span id="cb11-16"><a href="#cb11-16" aria-hidden="true" tabindex="-1"></a>  <span class="at">subtitle =</span> <span class="cn">NULL</span>,</span>
<span id="cb11-17"><a href="#cb11-17" aria-hidden="true" tabindex="-1"></a>  <span class="at">pCutoff =</span> <span class="fl">0.05</span>,</span>
<span id="cb11-18"><a href="#cb11-18" aria-hidden="true" tabindex="-1"></a>  <span class="at">pointSize =</span> <span class="dv">3</span>,</span>
<span id="cb11-19"><a href="#cb11-19" aria-hidden="true" tabindex="-1"></a>  <span class="at">labSize =</span> <span class="dv">3</span>,</span>
<span id="cb11-20"><a href="#cb11-20" aria-hidden="true" tabindex="-1"></a>  <span class="at">col =</span> <span class="fu">c</span>(<span class="st">"grey50"</span>, <span class="st">"dodgerblue3"</span>, <span class="st">"dodgerblue3"</span>, <span class="st">"firebrick2"</span>),</span>
<span id="cb11-21"><a href="#cb11-21" aria-hidden="true" tabindex="-1"></a>  <span class="at">legendLabels =</span> <span class="fu">c</span>(<span class="st">"NS"</span>, </span>
<span id="cb11-22"><a href="#cb11-22" aria-hidden="true" tabindex="-1"></a>                   <span class="fu">expression</span>(<span class="fu">abs</span>(log[<span class="dv">2</span>] <span class="sc">~</span> FC) <span class="sc">&gt;</span> <span class="dv">1</span>),</span>
<span id="cb11-23"><a href="#cb11-23" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"FDR &lt; 0.05"</span>,</span>
<span id="cb11-24"><a href="#cb11-24" aria-hidden="true" tabindex="-1"></a>                   <span class="fu">expression</span>(Pvalue <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">~</span> and <span class="sc">~</span> <span class="fu">abs</span>(log[<span class="dv">2</span>] <span class="sc">~</span> FC) <span class="sc">&gt;</span> <span class="dv">1</span>))) <span class="sc">+</span></span>
<span id="cb11-25"><a href="#cb11-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb11-26"><a href="#cb11-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"bottom"</span>,</span>
<span id="cb11-27"><a href="#cb11-27" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.border =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="cn">NA</span>, <span class="at">color =</span> <span class="st">"black"</span>),</span>
<span id="cb11-28"><a href="#cb11-28" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.grid =</span> <span class="fu">element_blank</span>())</span>
<span id="cb11-29"><a href="#cb11-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-30"><a href="#cb11-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-31"><a href="#cb11-31" aria-hidden="true" tabindex="-1"></a>p_volcano</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="plotting_omics_data_files/figure-html/unnamed-chunk-14-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<blockquote class="blockquote">
<p>See more <code>EnhancedVolcano()</code> usage: <a href="https://bioconductor.org/packages/devel/bioc/vignettes/EnhancedVolcano/inst/doc/EnhancedVolcano.html" class="uri">https://bioconductor.org/packages/devel/bioc/vignettes/EnhancedVolcano/inst/doc/EnhancedVolcano.html</a></p>
</blockquote>
</section>
<section id="venn-diagram-and-upset-plot" class="level2" data-number="6">
<h2 data-number="6" class="anchored" data-anchor-id="venn-diagram-and-upset-plot"><span class="header-section-number">6</span> Venn Diagram and Upset Plot</h2>
<section id="plotting-venn-diagram-with-ggvenn" class="level3" data-number="6.1">
<h3 data-number="6.1" class="anchored" data-anchor-id="plotting-venn-diagram-with-ggvenn"><span class="header-section-number">6.1</span> Plotting Venn diagram with <code>ggvenn</code></h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load data</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>mouse_hallmarks <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(<span class="st">"https://github.com/JirathNuan/r-handviz-workshop/raw/main/datasets/mouse_msigdb_6hallmarks.txt"</span>)</span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Split into list</span></span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a>lst_mouse_hallmark <span class="ot">&lt;-</span> <span class="fu">as.list</span>(mouse_hallmarks)</span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a><span class="co"># remove NAs</span></span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a>lst_mouse_hallmark <span class="ot">&lt;-</span> <span class="fu">lapply</span>(lst_mouse_hallmark, na.omit)</span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Chart</span></span>
<span id="cb12-10"><a href="#cb12-10" aria-hidden="true" tabindex="-1"></a><span class="fu">ggvenn</span>(</span>
<span id="cb12-11"><a href="#cb12-11" aria-hidden="true" tabindex="-1"></a>  lst_mouse_hallmark, </span>
<span id="cb12-12"><a href="#cb12-12" aria-hidden="true" tabindex="-1"></a>  <span class="at">fill_color =</span> <span class="fu">pal_npg</span>()(<span class="dv">6</span>),</span>
<span id="cb12-13"><a href="#cb12-13" aria-hidden="true" tabindex="-1"></a>  <span class="at">stroke_size =</span> <span class="fl">0.5</span>, </span>
<span id="cb12-14"><a href="#cb12-14" aria-hidden="true" tabindex="-1"></a>  <span class="at">set_name_size =</span> <span class="dv">3</span></span>
<span id="cb12-15"><a href="#cb12-15" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="plotting_omics_data_files/figure-html/unnamed-chunk-15-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="callout callout-style-simple callout-caution callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Venn diagram limitation
</div>
</div>
<div class="callout-body-container callout-body">
<p>Venn diagrams are simple and easy to interpret, however, ggvenn only allows the visualization of four features at a time. As a result, an alternative of Venn diagram has been developed.</p>
</div>
</div>
</section>
<section id="venn-diagram-alternative-upset-plot" class="level3" data-number="6.2">
<h3 data-number="6.2" class="anchored" data-anchor-id="venn-diagram-alternative-upset-plot"><span class="header-section-number">6.2</span> Venn diagram alternative: Upset Plot</h3>
<p>UpSet plots are more efficient than Venn Diagrams at visualizing intersections of multiple sets. They are particularly useful when analyzing large datasets with many variables. They help to identify patterns and relationships between different sets more quickly and easily than other visualizations.</p>
<p>There are 3 different modes of upset plot, distinct (default), intersect, and union. Here we’ll only demonstrate the default mode.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">123</span>)</span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Make present-absent matrix of data list</span></span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a>mouse_hm_mat01 <span class="ot">&lt;-</span> <span class="fu">list_to_matrix</span>(lst_mouse_hallmark)</span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Show how the combination matrix looks like</span></span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(mouse_hm_mat01) <span class="sc">%&gt;%</span> <span class="fu">kbl</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<table data-quarto-postprocess="true" class="table table-sm table-striped small">
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th"></th>
<th style="text-align: right;" data-quarto-table-cell-role="th">G2M checkpoint</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">Apoptosis</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">DNA Repair</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">WNT/B-Catenin Signaling</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">Hedgehog Signaling</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">TGF_BETA_SIGNALING</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Aaas</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
</tr>
<tr class="even">
<td style="text-align: left;">Abl1</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Ache</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0</td>
</tr>
<tr class="even">
<td style="text-align: left;">Acvr1</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Ada</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
</tr>
<tr class="even">
<td style="text-align: left;">Adam17</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
</tr>
</tbody>
</table>


</div>
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Make the combination matrix</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>comb_mat <span class="ot">&lt;-</span> <span class="fu">make_comb_mat</span>(lst_mouse_hallmark)</span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>comb_mat</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>A combination matrix with 6 sets and 18 combinations.
  ranges of combination set size: c(1, 183).
  mode for the combination size: distinct.
  sets are on rows.

Top 8 combination sets are:
  G2M checkpoint Apoptosis DNA Repair WNT/B-Catenin Signaling Hedgehog Signaling TGF_BETA_SIGNALING   code size
               x                                                                                    100000  183
                         x                                                                          010000  147
                                    x                                                               001000  143
                                                                                                  x 000001   45
                                                            x                                       000100   31
                                                                               x                    000010   31
               x         x                                                                          110000    4
                         x                                                                        x 010001    4

Sets are:
                      set size
           G2M checkpoint  195
                Apoptosis  161
               DNA Repair  148
  WNT/B-Catenin Signaling   42
       Hedgehog Signaling   36
       TGF_BETA_SIGNALING   53</code></pre>
</div>
</div>
<p><u>Referred from</u> a full guide of complexHeatmap: The UpSet plot visualizes the <strong>size</strong> of each combination set. With the binary code of each combination set, we can calculate the size. There are three modes.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="https://jokergoo.github.io/ComplexHeatmap-reference/book/08-upset_files/figure-html/unnamed-chunk-7-1.png" class="img-fluid figure-img" style="width:50.0%"></p>
<figcaption class="figure-caption">Figure from: <a href="https://jokergoo.github.io/ComplexHeatmap-reference/book/upset-plot.html#upset-plot" class="uri">https://jokergoo.github.io/ComplexHeatmap-reference/book/upset-plot.html#upset-plot</a></figcaption>
</figure>
</div>
<p>Plot upset plot</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>pal_upset <span class="ot">&lt;-</span> <span class="fu">pal_npg</span>()(<span class="dv">4</span>)</span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot upset following the default upset mode.</span></span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>upset_mouse_hm <span class="ot">&lt;-</span></span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">UpSet</span>(</span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a>    comb_mat,</span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">comb_col =</span> pal_upset[<span class="dv">1</span>],</span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">row_names_gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">10</span>),</span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">right_annotation =</span> <span class="fu">upset_right_annotation</span>(</span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a>      comb_mat,</span>
<span id="cb16-10"><a href="#cb16-10" aria-hidden="true" tabindex="-1"></a>      <span class="at">gp =</span> <span class="fu">gpar</span>(<span class="at">fill =</span> pal_upset[<span class="dv">2</span>]),</span>
<span id="cb16-11"><a href="#cb16-11" aria-hidden="true" tabindex="-1"></a>      <span class="at">add_numbers =</span> <span class="cn">TRUE</span></span>
<span id="cb16-12"><a href="#cb16-12" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb16-13"><a href="#cb16-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">top_annotation =</span> <span class="fu">upset_top_annotation</span>(</span>
<span id="cb16-14"><a href="#cb16-14" aria-hidden="true" tabindex="-1"></a>      comb_mat,</span>
<span id="cb16-15"><a href="#cb16-15" aria-hidden="true" tabindex="-1"></a>      <span class="at">gp =</span> <span class="fu">gpar</span>(<span class="at">fill =</span> pal_upset[<span class="dv">1</span>]),</span>
<span id="cb16-16"><a href="#cb16-16" aria-hidden="true" tabindex="-1"></a>      <span class="at">add_numbers =</span> <span class="cn">TRUE</span></span>
<span id="cb16-17"><a href="#cb16-17" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb16-18"><a href="#cb16-18" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb16-19"><a href="#cb16-19" aria-hidden="true" tabindex="-1"></a>upset_mouse_hm</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="plotting_omics_data_files/figure-html/unnamed-chunk-17-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<blockquote class="blockquote">
<p>See more <code>UpSet()</code> usage from a complete reference of complexHeatmap: <a href="https://jokergoo.github.io/ComplexHeatmap-reference/book/upset-plot.html#upset-plot" class="uri">https://jokergoo.github.io/ComplexHeatmap-reference/book/upset-plot.html#upset-plot</a></p>
</blockquote>
</section>
</section>
<section id="ggplot2-based-publication-ready-plots-with-ggpubr" class="level2" data-number="7">
<h2 data-number="7" class="anchored" data-anchor-id="ggplot2-based-publication-ready-plots-with-ggpubr"><span class="header-section-number">7</span> ggplot2 Based Publication Ready Plots with <code>ggpubr</code></h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load dataset</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>mouseLiver_ctg <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">"https://github.com/JirathNuan/r-handviz-workshop/raw/main/datasets/mouseLiver_data_ClinicalTraits.csv"</span>)</span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Inspect the data</span></span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(mouseLiver_ctg) <span class="sc">%&gt;%</span> <span class="fu">kbl</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<table data-quarto-postprocess="true" class="table table-sm table-striped small">
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th">Mice</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Strain</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">sex</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">weight_g</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">length_cm</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">ab_fat</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">other_fat</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">total_fat</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">100xfat_weight</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">Trigly</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">Total_Chol</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">HDL_Chol</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">UC</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">FFA</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">Glucose</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">LDL_plus_VLDL</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">MCP_1_phys</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">Insulin_ug_l</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">Glucose_Insulin</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">Leptin_pg_ml</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">Adiponectin</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">Aortic lesions</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">F2_290</td>
<td style="text-align: left;">BxH ApoE-/-, F2</td>
<td style="text-align: left;">Female</td>
<td style="text-align: right;">36.9</td>
<td style="text-align: right;">9.9</td>
<td style="text-align: right;">2.53</td>
<td style="text-align: right;">2.26</td>
<td style="text-align: right;">4.79</td>
<td style="text-align: right;">12.981030</td>
<td style="text-align: right;">53</td>
<td style="text-align: right;">1167</td>
<td style="text-align: right;">50</td>
<td style="text-align: right;">484</td>
<td style="text-align: right;">121</td>
<td style="text-align: right;">437</td>
<td style="text-align: right;">1117</td>
<td style="text-align: right;">175.850</td>
<td style="text-align: right;">924</td>
<td style="text-align: right;">0.4729437</td>
<td style="text-align: right;">245462.00</td>
<td style="text-align: right;">11.274</td>
<td style="text-align: right;">496250</td>
</tr>
<tr class="even">
<td style="text-align: left;">F2_291</td>
<td style="text-align: left;">BxH ApoE-/-, F2</td>
<td style="text-align: left;">Female</td>
<td style="text-align: right;">48.5</td>
<td style="text-align: right;">10.7</td>
<td style="text-align: right;">2.90</td>
<td style="text-align: right;">2.97</td>
<td style="text-align: right;">5.87</td>
<td style="text-align: right;">12.103093</td>
<td style="text-align: right;">61</td>
<td style="text-align: right;">1230</td>
<td style="text-align: right;">32</td>
<td style="text-align: right;">592</td>
<td style="text-align: right;">173</td>
<td style="text-align: right;">572</td>
<td style="text-align: right;">1198</td>
<td style="text-align: right;">92.430</td>
<td style="text-align: right;">5781</td>
<td style="text-align: right;">0.0989448</td>
<td style="text-align: right;">84420.88</td>
<td style="text-align: right;">7.099</td>
<td style="text-align: right;">NA</td>
</tr>
<tr class="odd">
<td style="text-align: left;">F2_292</td>
<td style="text-align: left;">BxH ApoE-/-, F2</td>
<td style="text-align: left;">Male</td>
<td style="text-align: right;">45.7</td>
<td style="text-align: right;">10.4</td>
<td style="text-align: right;">1.04</td>
<td style="text-align: right;">2.31</td>
<td style="text-align: right;">3.35</td>
<td style="text-align: right;">7.330416</td>
<td style="text-align: right;">41</td>
<td style="text-align: right;">1285</td>
<td style="text-align: right;">81</td>
<td style="text-align: right;">460</td>
<td style="text-align: right;">96</td>
<td style="text-align: right;">497</td>
<td style="text-align: right;">1204</td>
<td style="text-align: right;">196.398</td>
<td style="text-align: right;">2074</td>
<td style="text-align: right;">0.2396336</td>
<td style="text-align: right;">105889.76</td>
<td style="text-align: right;">5.795</td>
<td style="text-align: right;">218500</td>
</tr>
<tr class="even">
<td style="text-align: left;">F2_293</td>
<td style="text-align: left;">BxH ApoE-/-, F2</td>
<td style="text-align: left;">Male</td>
<td style="text-align: right;">50.3</td>
<td style="text-align: right;">10.9</td>
<td style="text-align: right;">0.91</td>
<td style="text-align: right;">1.89</td>
<td style="text-align: right;">2.80</td>
<td style="text-align: right;">5.566600</td>
<td style="text-align: right;">271</td>
<td style="text-align: right;">1299</td>
<td style="text-align: right;">64</td>
<td style="text-align: right;">476</td>
<td style="text-align: right;">122</td>
<td style="text-align: right;">553</td>
<td style="text-align: right;">1235</td>
<td style="text-align: right;">97.466</td>
<td style="text-align: right;">11874</td>
<td style="text-align: right;">0.0465723</td>
<td style="text-align: right;">100398.68</td>
<td style="text-align: right;">5.495</td>
<td style="text-align: right;">61250</td>
</tr>
<tr class="odd">
<td style="text-align: left;">F2_294</td>
<td style="text-align: left;">BxH ApoE-/-, F2</td>
<td style="text-align: left;">Male</td>
<td style="text-align: right;">44.8</td>
<td style="text-align: right;">9.8</td>
<td style="text-align: right;">1.22</td>
<td style="text-align: right;">2.47</td>
<td style="text-align: right;">3.69</td>
<td style="text-align: right;">8.236607</td>
<td style="text-align: right;">114</td>
<td style="text-align: right;">1410</td>
<td style="text-align: right;">50</td>
<td style="text-align: right;">516</td>
<td style="text-align: right;">118</td>
<td style="text-align: right;">535</td>
<td style="text-align: right;">1360</td>
<td style="text-align: right;">95.452</td>
<td style="text-align: right;">9181</td>
<td style="text-align: right;">0.0582725</td>
<td style="text-align: right;">130846.30</td>
<td style="text-align: right;">6.868</td>
<td style="text-align: right;">243750</td>
</tr>
<tr class="even">
<td style="text-align: left;">F2_295</td>
<td style="text-align: left;">BxH ApoE-/-, F2</td>
<td style="text-align: left;">Male</td>
<td style="text-align: right;">39.2</td>
<td style="text-align: right;">10.2</td>
<td style="text-align: right;">3.06</td>
<td style="text-align: right;">2.49</td>
<td style="text-align: right;">5.55</td>
<td style="text-align: right;">14.158163</td>
<td style="text-align: right;">72</td>
<td style="text-align: right;">1533</td>
<td style="text-align: right;">18</td>
<td style="text-align: right;">620</td>
<td style="text-align: right;">106</td>
<td style="text-align: right;">382</td>
<td style="text-align: right;">1515</td>
<td style="text-align: right;">144.270</td>
<td style="text-align: right;">485</td>
<td style="text-align: right;">0.7876289</td>
<td style="text-align: right;">75166.22</td>
<td style="text-align: right;">17.328</td>
<td style="text-align: right;">104250</td>
</tr>
</tbody>
</table>


</div>
</div>
<section id="histogram" class="level3" data-number="7.1">
<h3 data-number="7.1" class="anchored" data-anchor-id="histogram"><span class="header-section-number">7.1</span> Histogram</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="fu">gghistogram</span>(</span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>  mouseLiver_ctg,</span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">x =</span> <span class="st">"weight_g"</span>,</span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">add =</span> <span class="st">"mean"</span>,</span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">rug =</span> <span class="cn">TRUE</span>,</span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">color =</span> <span class="st">"sex"</span>,</span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">fill =</span> <span class="st">"sex"</span>,</span>
<span id="cb18-8"><a href="#cb18-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">bins =</span> <span class="dv">30</span>,</span>
<span id="cb18-9"><a href="#cb18-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">palette =</span> <span class="st">"npg"</span>,</span>
<span id="cb18-10"><a href="#cb18-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">xlab =</span> <span class="st">"Weight (g)"</span>,</span>
<span id="cb18-11"><a href="#cb18-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">ylab =</span> <span class="st">"Frequency"</span></span>
<span id="cb18-12"><a href="#cb18-12" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="plotting_omics_data_files/figure-html/unnamed-chunk-19-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="density-plot" class="level3" data-number="7.2">
<h3 data-number="7.2" class="anchored" data-anchor-id="density-plot"><span class="header-section-number">7.2</span> Density plot</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggdensity</span>(</span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>  mouseLiver_ctg,</span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">x =</span> <span class="st">"weight_g"</span>,</span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">add =</span> <span class="st">"mean"</span>,</span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">rug =</span> <span class="cn">TRUE</span>,</span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">color =</span> <span class="st">"sex"</span>,</span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">fill =</span> <span class="st">"sex"</span>,</span>
<span id="cb19-8"><a href="#cb19-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">palette =</span> <span class="st">"npg"</span>,</span>
<span id="cb19-9"><a href="#cb19-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">xlab =</span> <span class="st">"Weight (g)"</span>,</span>
<span id="cb19-10"><a href="#cb19-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">ylab =</span> <span class="st">"Frequency"</span></span>
<span id="cb19-11"><a href="#cb19-11" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="plotting_omics_data_files/figure-html/unnamed-chunk-20-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="simple-box-plot" class="level3" data-number="7.3">
<h3 data-number="7.3" class="anchored" data-anchor-id="simple-box-plot"><span class="header-section-number">7.3</span> Simple box plot</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a>p_boxplot <span class="ot">&lt;-</span> <span class="fu">ggboxplot</span>(ToothGrowth, <span class="at">x =</span> <span class="st">"supp"</span>, <span class="at">y =</span> <span class="st">"len"</span>,</span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a>          <span class="at">color =</span> <span class="st">"supp"</span>, <span class="at">palette =</span> <span class="st">"npg"</span>,</span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a>          <span class="at">add =</span> <span class="st">"jitter"</span>,</span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a>          <span class="at">xlab =</span> <span class="st">"Supplement type"</span>,</span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a>          <span class="at">ylab =</span> <span class="st">"Tooth length"</span>)</span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a>p_boxplot</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="plotting_omics_data_files/figure-html/unnamed-chunk-21-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>Adding statistical significance to the plot</p>
<ul>
<li>Pairwise comparison</li>
</ul>
<div>
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Try to calculate stat first</span></span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a><span class="fu">compare_means</span>(len <span class="sc">~</span> supp, <span class="at">data =</span> ToothGrowth, <span class="at">method =</span> <span class="st">"anova"</span>)</span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a><span class="fu">compare_means</span>(len <span class="sc">~</span> supp, <span class="at">data =</span> ToothGrowth, <span class="at">method =</span> <span class="st">"wilcox.test"</span>)</span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot box plot with significance</span></span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a>p_boxplot <span class="sc">+</span> <span class="fu">stat_compare_means</span>(<span class="at">method =</span> <span class="st">"anova"</span>)</span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a>p_boxplot <span class="sc">+</span> <span class="fu">stat_compare_means</span>(<span class="at">method =</span> <span class="st">"wilcox.test"</span>)</span>
<span id="cb21-7"><a href="#cb21-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Change style of singnificance notation</span></span>
<span id="cb21-8"><a href="#cb21-8" aria-hidden="true" tabindex="-1"></a>p_boxplot <span class="sc">+</span> <span class="fu">stat_compare_means</span>(<span class="at">method =</span> <span class="st">"anova"</span>,</span>
<span id="cb21-9"><a href="#cb21-9" aria-hidden="true" tabindex="-1"></a>                               <span class="fu">aes</span>(<span class="at">label =</span> ..p.signif..),</span>
<span id="cb21-10"><a href="#cb21-10" aria-hidden="true" tabindex="-1"></a>                               <span class="at">label.x =</span> <span class="fl">1.5</span>,</span>
<span id="cb21-11"><a href="#cb21-11" aria-hidden="true" tabindex="-1"></a>                               <span class="at">label.y =</span> <span class="dv">40</span>)</span>
<span id="cb21-12"><a href="#cb21-12" aria-hidden="true" tabindex="-1"></a>p_boxplot <span class="sc">+</span> <span class="fu">stat_compare_means</span>(<span class="at">method =</span> <span class="st">"wilcox.test"</span>,</span>
<span id="cb21-13"><a href="#cb21-13" aria-hidden="true" tabindex="-1"></a>                               <span class="fu">aes</span>(<span class="at">label =</span> <span class="st">"p.signif"</span>),</span>
<span id="cb21-14"><a href="#cb21-14" aria-hidden="true" tabindex="-1"></a>                               <span class="at">label.x =</span> <span class="fl">1.5</span>,</span>
<span id="cb21-15"><a href="#cb21-15" aria-hidden="true" tabindex="-1"></a>                               <span class="at">label.y =</span> <span class="dv">40</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell quarto-layout-panel">
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="cell-output cell-output-stdout quarto-layout-cell" style="flex-basis: 50.0%;justify-content: center;">
<pre><code># A tibble: 1 × 6
  .y.        p p.adj p.format p.signif method
  &lt;chr&gt;  &lt;dbl&gt; &lt;dbl&gt; &lt;chr&gt;    &lt;chr&gt;    &lt;chr&gt; 
1 len   0.0604  0.06 0.06     ns       Anova </code></pre>
</div>
<div class="cell-output cell-output-stdout quarto-layout-cell" style="flex-basis: 50.0%;justify-content: center;">
<pre><code># A tibble: 1 × 8
  .y.   group1 group2      p p.adj p.format p.signif method  
  &lt;chr&gt; &lt;chr&gt;  &lt;chr&gt;   &lt;dbl&gt; &lt;dbl&gt; &lt;chr&gt;    &lt;chr&gt;    &lt;chr&gt;   
1 len   OJ     VC     0.0645 0.064 0.064    ns       Wilcoxon</code></pre>
</div>
</div>
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="cell-output-display quarto-layout-cell" style="flex-basis: 50.0%;justify-content: center;">
<p><img src="plotting_omics_data_files/figure-html/unnamed-chunk-22-1.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display quarto-layout-cell" style="flex-basis: 50.0%;justify-content: center;">
<p><img src="plotting_omics_data_files/figure-html/unnamed-chunk-22-2.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="cell-output-display quarto-layout-cell" style="flex-basis: 50.0%;justify-content: center;">
<p><img src="plotting_omics_data_files/figure-html/unnamed-chunk-22-3.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display quarto-layout-cell" style="flex-basis: 50.0%;justify-content: center;">
<p><img src="plotting_omics_data_files/figure-html/unnamed-chunk-22-4.png" class="img-fluid" width="672"></p>
</div>
</div>
</div>
</div>
<ul>
<li>Compare more than two groups</li>
</ul>
<div>
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Global test</span></span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a><span class="fu">compare_means</span>(len <span class="sc">~</span> dose,  <span class="at">data =</span> ToothGrowth, <span class="at">method =</span> <span class="st">"anova"</span>)</span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Default method = "kruskal.test" for multiple groups</span></span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a><span class="fu">ggboxplot</span>(ToothGrowth, <span class="at">x =</span> <span class="st">"dose"</span>, <span class="at">y =</span> <span class="st">"len"</span>,</span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a>          <span class="at">fill =</span> <span class="st">"dose"</span>, <span class="at">palette =</span> <span class="st">"npg"</span>)<span class="sc">+</span></span>
<span id="cb24-6"><a href="#cb24-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_compare_means</span>()</span>
<span id="cb24-7"><a href="#cb24-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Change method to anova</span></span>
<span id="cb24-8"><a href="#cb24-8" aria-hidden="true" tabindex="-1"></a><span class="fu">ggboxplot</span>(ToothGrowth, <span class="at">x =</span> <span class="st">"dose"</span>, <span class="at">y =</span> <span class="st">"len"</span>,</span>
<span id="cb24-9"><a href="#cb24-9" aria-hidden="true" tabindex="-1"></a>          <span class="at">fill =</span> <span class="st">"dose"</span>, <span class="at">palette =</span> <span class="st">"npg"</span>)<span class="sc">+</span></span>
<span id="cb24-10"><a href="#cb24-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_compare_means</span>(<span class="at">method =</span> <span class="st">"anova"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell quarto-layout-panel">
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="cell-output cell-output-stdout quarto-layout-cell" style="flex-basis: 50.0%;justify-content: center;">
<pre><code># A tibble: 1 × 6
  .y.          p   p.adj p.format p.signif method
  &lt;chr&gt;    &lt;dbl&gt;   &lt;dbl&gt; &lt;chr&gt;    &lt;chr&gt;    &lt;chr&gt; 
1 len   9.53e-16 9.5e-16 9.5e-16  ****     Anova </code></pre>
</div>
<div class="cell-output-display quarto-layout-cell" style="flex-basis: 50.0%;justify-content: center;">
<p><img src="plotting_omics_data_files/figure-html/unnamed-chunk-23-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="cell-output-display quarto-layout-cell" style="flex-basis: 50.0%;justify-content: center;">
<p><img src="plotting_omics_data_files/figure-html/unnamed-chunk-23-2.png" class="img-fluid" width="672"></p>
</div>
</div>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Perorm pairwise comparisons</span></span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a><span class="fu">compare_means</span>(len <span class="sc">~</span> dose,  <span class="at">data =</span> ToothGrowth)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 3 × 8
  .y.   group1 group2            p      p.adj p.format p.signif method  
  &lt;chr&gt; &lt;chr&gt;  &lt;chr&gt;         &lt;dbl&gt;      &lt;dbl&gt; &lt;chr&gt;    &lt;chr&gt;    &lt;chr&gt;   
1 len   0.5    1      0.00000702   0.000014   7.0e-06  ****     Wilcoxon
2 len   0.5    2      0.0000000841 0.00000025 8.4e-08  ****     Wilcoxon
3 len   1      2      0.000177     0.00018    0.00018  ***      Wilcoxon</code></pre>
</div>
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize: Specify the comparisons you want</span></span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a>my_comparisons <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="fu">c</span>(<span class="st">"0.5"</span>, <span class="st">"1"</span>), <span class="fu">c</span>(<span class="st">"1"</span>, <span class="st">"2"</span>), <span class="fu">c</span>(<span class="st">"0.5"</span>, <span class="st">"2"</span>))</span>
<span id="cb28-3"><a href="#cb28-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-4"><a href="#cb28-4" aria-hidden="true" tabindex="-1"></a><span class="fu">ggboxplot</span>(</span>
<span id="cb28-5"><a href="#cb28-5" aria-hidden="true" tabindex="-1"></a>  ToothGrowth,</span>
<span id="cb28-6"><a href="#cb28-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">x =</span> <span class="st">"dose"</span>,</span>
<span id="cb28-7"><a href="#cb28-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">y =</span> <span class="st">"len"</span>,</span>
<span id="cb28-8"><a href="#cb28-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">fill =</span> <span class="st">"dose"</span>,</span>
<span id="cb28-9"><a href="#cb28-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">palette =</span> <span class="st">"npg"</span>) <span class="sc">+</span></span>
<span id="cb28-10"><a href="#cb28-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_compare_means</span>(<span class="at">comparisons =</span> my_comparisons) <span class="sc">+</span> <span class="co"># Add pairwise comparisons p-value</span></span>
<span id="cb28-11"><a href="#cb28-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_compare_means</span>(<span class="at">label.y =</span> <span class="dv">50</span>)     <span class="co"># Add global p-value</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="plotting_omics_data_files/figure-html/unnamed-chunk-24-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<ul>
<li>Multiple pairwise tests against a reference group</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Pairwise comparison against reference</span></span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a><span class="fu">compare_means</span>(len <span class="sc">~</span> dose,</span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a>              <span class="at">data =</span> ToothGrowth,</span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a>              <span class="at">ref.group =</span> <span class="st">"0.5"</span>,</span>
<span id="cb29-5"><a href="#cb29-5" aria-hidden="true" tabindex="-1"></a>              <span class="at">method =</span> <span class="st">"t.test"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 2 × 8
  .y.   group1 group2        p    p.adj p.format p.signif method
  &lt;chr&gt; &lt;chr&gt;  &lt;chr&gt;     &lt;dbl&gt;    &lt;dbl&gt; &lt;chr&gt;    &lt;chr&gt;    &lt;chr&gt; 
1 len   0.5    1      1.27e- 7 1.3 e- 7 1.3e-07  ****     T-test
2 len   0.5    2      4.40e-14 8.80e-14 4.4e-14  ****     T-test</code></pre>
</div>
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize</span></span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggboxplot</span>(</span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a>  ToothGrowth,</span>
<span id="cb31-4"><a href="#cb31-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">x =</span> <span class="st">"dose"</span>,</span>
<span id="cb31-5"><a href="#cb31-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">y =</span> <span class="st">"len"</span>,</span>
<span id="cb31-6"><a href="#cb31-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">fill =</span> <span class="st">"dose"</span>,</span>
<span id="cb31-7"><a href="#cb31-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">palette =</span> <span class="st">"npg"</span>) <span class="sc">+</span></span>
<span id="cb31-8"><a href="#cb31-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_compare_means</span>(<span class="at">method =</span> <span class="st">"anova"</span>, <span class="at">label.y =</span> <span class="dv">40</span>) <span class="sc">+</span>      <span class="co"># Add global p-value</span></span>
<span id="cb31-9"><a href="#cb31-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_compare_means</span>(<span class="at">label =</span> <span class="st">"p.signif"</span>,</span>
<span id="cb31-10"><a href="#cb31-10" aria-hidden="true" tabindex="-1"></a>                     <span class="at">method =</span> <span class="st">"t.test"</span>,</span>
<span id="cb31-11"><a href="#cb31-11" aria-hidden="true" tabindex="-1"></a>                     <span class="at">ref.group =</span> <span class="st">"0.5"</span>)                    <span class="co"># Pairwise comparison against reference</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="plotting_omics_data_files/figure-html/unnamed-chunk-25-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>Multiple grouping variables</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="fu">compare_means</span>(len <span class="sc">~</span> supp, <span class="at">data =</span> ToothGrowth, <span class="at">group.by =</span> <span class="st">"dose"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 3 × 9
   dose .y.   group1 group2       p p.adj p.format p.signif method  
  &lt;dbl&gt; &lt;chr&gt; &lt;chr&gt;  &lt;chr&gt;    &lt;dbl&gt; &lt;dbl&gt; &lt;chr&gt;    &lt;chr&gt;    &lt;chr&gt;   
1   0.5 len   OJ     VC     0.0232  0.046 0.023    *        Wilcoxon
2   1   len   OJ     VC     0.00403 0.012 0.004    **       Wilcoxon
3   2   len   OJ     VC     1       1     1.000    ns       Wilcoxon</code></pre>
</div>
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Box plot facetted by "dose"</span></span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a>p_box <span class="ot">&lt;-</span> <span class="fu">ggboxplot</span>(</span>
<span id="cb34-3"><a href="#cb34-3" aria-hidden="true" tabindex="-1"></a>  ToothGrowth,</span>
<span id="cb34-4"><a href="#cb34-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">x =</span> <span class="st">"supp"</span>,</span>
<span id="cb34-5"><a href="#cb34-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">y =</span> <span class="st">"len"</span>,</span>
<span id="cb34-6"><a href="#cb34-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">color =</span> <span class="st">"supp"</span>,</span>
<span id="cb34-7"><a href="#cb34-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">palette =</span> <span class="st">"jco"</span>,</span>
<span id="cb34-8"><a href="#cb34-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">add =</span> <span class="st">"jitter"</span>,</span>
<span id="cb34-9"><a href="#cb34-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">facet.by =</span> <span class="st">"dose"</span>,</span>
<span id="cb34-10"><a href="#cb34-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">short.panel.labs =</span> <span class="cn">FALSE</span></span>
<span id="cb34-11"><a href="#cb34-11" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb34-12"><a href="#cb34-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Use only p.format as label. Remove method name.</span></span>
<span id="cb34-13"><a href="#cb34-13" aria-hidden="true" tabindex="-1"></a>p_box <span class="sc">+</span> <span class="fu">stat_compare_means</span>(<span class="at">label =</span> <span class="st">"p.format"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="plotting_omics_data_files/figure-html/unnamed-chunk-26-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="bar-and-line-plots-one-grouping-variable" class="level3" data-number="7.4">
<h3 data-number="7.4" class="anchored" data-anchor-id="bar-and-line-plots-one-grouping-variable"><span class="header-section-number">7.4</span> Bar and line plots (one grouping variable):</h3>
<div>
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggbarplot</span>(</span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a>  ToothGrowth,</span>
<span id="cb35-3"><a href="#cb35-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">x =</span> <span class="st">"dose"</span>,</span>
<span id="cb35-4"><a href="#cb35-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">y =</span> <span class="st">"len"</span>,</span>
<span id="cb35-5"><a href="#cb35-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">add =</span> <span class="st">"mean_se"</span>,</span>
<span id="cb35-6"><a href="#cb35-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">fill =</span> <span class="st">"supp"</span>,</span>
<span id="cb35-7"><a href="#cb35-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">palette =</span> <span class="st">"npg"</span>,</span>
<span id="cb35-8"><a href="#cb35-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">color =</span> <span class="st">"black"</span>,</span>
<span id="cb35-9"><a href="#cb35-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="fl">0.8</span>)) <span class="sc">+</span></span>
<span id="cb35-10"><a href="#cb35-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_compare_means</span>(<span class="fu">aes</span>(<span class="at">group =</span> supp), <span class="at">label =</span> <span class="st">"p.signif"</span>, <span class="at">label.y =</span> <span class="dv">29</span>)</span>
<span id="cb35-11"><a href="#cb35-11" aria-hidden="true" tabindex="-1"></a><span class="fu">ggline</span>(</span>
<span id="cb35-12"><a href="#cb35-12" aria-hidden="true" tabindex="-1"></a>  ToothGrowth,</span>
<span id="cb35-13"><a href="#cb35-13" aria-hidden="true" tabindex="-1"></a>  <span class="at">x =</span> <span class="st">"dose"</span>,</span>
<span id="cb35-14"><a href="#cb35-14" aria-hidden="true" tabindex="-1"></a>  <span class="at">y =</span> <span class="st">"len"</span>,</span>
<span id="cb35-15"><a href="#cb35-15" aria-hidden="true" tabindex="-1"></a>  <span class="at">add =</span> <span class="st">"mean_se"</span>,</span>
<span id="cb35-16"><a href="#cb35-16" aria-hidden="true" tabindex="-1"></a>  <span class="at">color =</span> <span class="st">"supp"</span>,</span>
<span id="cb35-17"><a href="#cb35-17" aria-hidden="true" tabindex="-1"></a>  <span class="at">palette =</span> <span class="st">"npg"</span>) <span class="sc">+</span></span>
<span id="cb35-18"><a href="#cb35-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_compare_means</span>(<span class="fu">aes</span>(<span class="at">group =</span> supp),</span>
<span id="cb35-19"><a href="#cb35-19" aria-hidden="true" tabindex="-1"></a>                     <span class="at">label =</span> <span class="st">"p.signif"</span>,</span>
<span id="cb35-20"><a href="#cb35-20" aria-hidden="true" tabindex="-1"></a>                     <span class="at">label.y =</span> <span class="fu">c</span>(<span class="dv">16</span>, <span class="dv">25</span>, <span class="dv">29</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell quarto-layout-panel">
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="cell-output-display quarto-layout-cell" style="flex-basis: 50.0%;justify-content: center;">
<p><img src="plotting_omics_data_files/figure-html/unnamed-chunk-27-1.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display quarto-layout-cell" style="flex-basis: 50.0%;justify-content: center;">
<p><img src="plotting_omics_data_files/figure-html/unnamed-chunk-27-2.png" class="img-fluid" width="672"></p>
</div>
</div>
</div>
</div>
</section>
<section id="scatter-plot" class="level3" data-number="7.5">
<h3 data-number="7.5" class="anchored" data-anchor-id="scatter-plot"><span class="header-section-number">7.5</span> Scatter plot</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a>p4 <span class="ot">&lt;-</span> <span class="fu">ggscatter</span>(</span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a>  mtcars,</span>
<span id="cb36-3"><a href="#cb36-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">x =</span> <span class="st">"wt"</span>,</span>
<span id="cb36-4"><a href="#cb36-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">y =</span> <span class="st">"mpg"</span>,</span>
<span id="cb36-5"><a href="#cb36-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">fill =</span> <span class="st">"mpg"</span>,</span>
<span id="cb36-6"><a href="#cb36-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">size =</span> <span class="dv">4</span>,</span>
<span id="cb36-7"><a href="#cb36-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">shape =</span> <span class="dv">21</span>) <span class="sc">+</span></span>
<span id="cb36-8"><a href="#cb36-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">gradient_fill</span>(<span class="fu">c</span>(<span class="st">"blue"</span>, <span class="st">"white"</span>, <span class="st">"red"</span>))</span>
<span id="cb36-9"><a href="#cb36-9" aria-hidden="true" tabindex="-1"></a>p4</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="plotting_omics_data_files/figure-html/unnamed-chunk-28-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<blockquote class="blockquote">
<p>More usage of ggpubr: <a href="http://www.sthda.com/english/articles/24-ggpubr-publication-ready-plots/76-add-p-values-and-significance-levels-to-ggplots/" class="uri">http://www.sthda.com/english/articles/24-ggpubr-publication-ready-plots/76-add-p-values-and-significance-levels-to-ggplots/</a>.</p>
</blockquote>
</section>
</section>
<section id="session-info" class="level2" data-number="8">
<h2 data-number="8" class="anchored" data-anchor-id="session-info"><span class="header-section-number">8</span> Session info</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>R version 4.3.0 (2023-04-21 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 11 x64 (build 22621)

Matrix products: default


locale:
[1] LC_COLLATE=English_United States.utf8 
[2] LC_CTYPE=English_United States.utf8   
[3] LC_MONETARY=English_United States.utf8
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.utf8    

time zone: Asia/Bangkok
tzcode source: internal

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] ggpubr_0.6.0                ggvenn_0.1.10              
 [3] EnhancedVolcano_1.18.0      ggrepel_0.9.3              
 [5] RColorBrewer_1.1-3          ComplexHeatmap_2.16.0      
 [7] ggsci_3.0.0                 factoextra_1.0.7           
 [9] pcaExplorer_2.26.1          DESeq2_1.40.1              
[11] SummarizedExperiment_1.30.1 Biobase_2.60.0             
[13] MatrixGenerics_1.12.0       matrixStats_0.63.0         
[15] GenomicRanges_1.52.0        GenomeInfoDb_1.36.0        
[17] IRanges_2.34.0              S4Vectors_0.38.1           
[19] BiocGenerics_0.46.0         openxlsx_4.2.5.2           
[21] kableExtra_1.3.4            lubridate_1.9.2            
[23] forcats_1.0.0               stringr_1.5.0              
[25] dplyr_1.1.2                 purrr_1.0.1                
[27] readr_2.1.4                 tidyr_1.3.0                
[29] tibble_3.2.1                ggplot2_3.4.2              
[31] tidyverse_2.0.0            

loaded via a namespace (and not attached):
  [1] splines_4.3.0           later_1.3.1             bitops_1.0-7           
  [4] filelock_1.0.2          graph_1.78.0            XML_3.99-0.14          
  [7] lifecycle_1.0.3         rstatix_0.7.2           topGO_2.52.0           
 [10] doParallel_1.0.17       vroom_1.6.3             lattice_0.21-8         
 [13] crosstalk_1.2.0         backports_1.4.1         dendextend_1.17.1      
 [16] magrittr_2.0.3          limma_3.56.1            plotly_4.10.1          
 [19] rmarkdown_2.21          yaml_2.3.7              shinyBS_0.61.1         
 [22] httpuv_1.6.10           NMF_0.26                zip_2.3.0              
 [25] DBI_1.1.3               abind_1.4-5             zlibbioc_1.46.0        
 [28] rvest_1.0.3             RCurl_1.98-1.12         rappdirs_0.3.3         
 [31] circlize_0.4.15         seriation_1.4.2         GenomeInfoDbData_1.2.10
 [34] AnnotationForge_1.42.0  genefilter_1.82.1       pheatmap_1.0.12        
 [37] annotate_1.78.0         svglite_2.1.1           codetools_0.2-19       
 [40] DelayedArray_0.26.2     DT_0.27                 xml2_1.3.4             
 [43] shape_1.4.6             tidyselect_1.2.0        GOstats_2.66.0         
 [46] farver_2.1.1            viridis_0.6.3           TSP_1.2-4              
 [49] BiocFileCache_2.8.0     base64enc_0.1-3         webshot_0.5.4          
 [52] jsonlite_1.8.4          GetoptLong_1.0.5        ellipsis_0.3.2         
 [55] survival_3.5-5          iterators_1.0.14        systemfonts_1.0.4      
 [58] foreach_1.5.2           tools_4.3.0             progress_1.2.2         
 [61] Rcpp_1.0.10             glue_1.6.2              gridExtra_2.3          
 [64] xfun_0.39               ca_0.71.1               shinydashboard_0.7.2   
 [67] withr_2.5.0             BiocManager_1.30.20     Category_2.66.0        
 [70] fastmap_1.1.1           fansi_1.0.4             SparseM_1.81           
 [73] digest_0.6.31           timechange_0.2.0        R6_2.5.1               
 [76] mime_0.12               colorspace_2.1-0        GO.db_3.17.0           
 [79] biomaRt_2.56.0          RSQLite_2.3.1           threejs_0.3.3          
 [82] utf8_1.2.3              generics_0.1.3          data.table_1.14.8      
 [85] prettyunits_1.1.1       httr_1.4.6              htmlwidgets_1.6.2      
 [88] S4Arrays_1.0.1          pkgconfig_2.0.3         gtable_0.3.3           
 [91] blob_1.2.4              registry_0.5-1          XVector_0.40.0         
 [94] htmltools_0.5.5         carData_3.0-5           RBGL_1.76.0            
 [97] clue_0.3-64             GSEABase_1.62.0         scales_1.2.1           
[100] png_0.1-8               knitr_1.42              rstudioapi_0.14        
[103] rjson_0.2.21            tzdb_0.3.0              reshape2_1.4.4         
[106] curl_5.0.0              shinyAce_0.4.2          GlobalOptions_0.1.2    
[109] cachem_1.0.8            parallel_4.3.0          AnnotationDbi_1.62.1   
[112] pillar_1.9.0            vctrs_0.6.2             promises_1.2.0.1       
[115] car_3.1-2               dbplyr_2.3.2            xtable_1.8-4           
[118] cluster_2.1.4           Rgraphviz_2.44.0        evaluate_0.21          
[121] cli_3.6.1               locfit_1.5-9.7          compiler_4.3.0         
[124] rlang_1.1.1             crayon_1.5.2            rngtools_1.5.2         
[127] ggsignif_0.6.4          heatmaply_1.4.2         labeling_0.4.2         
[130] plyr_1.8.8              stringi_1.7.12          viridisLite_0.4.2      
[133] gridBase_0.4-7          BiocParallel_1.34.1     assertthat_0.2.1       
[136] munsell_0.5.0           Biostrings_2.68.0       lazyeval_0.2.2         
[139] pacman_0.5.1            Matrix_1.5-4            hms_1.1.3              
[142] bit64_4.0.5             KEGGREST_1.40.0         shiny_1.7.4            
[145] highr_0.10              broom_1.0.4             igraph_1.4.2           
[148] memoise_2.0.1           bit_4.0.5              </code></pre>
</div>
</div>



</section>

<div id="quarto-appendix" class="default"><section class="quarto-appendix-contents" role="doc-bibliography"><h2 class="anchored quarto-appendix-heading">References</h2><div id="refs" class="references csl-bib-body hanging-indent" role="list">
<div id="ref-brooks2010" class="csl-entry" role="listitem">
Brooks, Angela N., Li Yang, Michael O. Duff, Kasper D. Hansen, Jung W. Park, Sandrine Dudoit, Steven E. Brenner, and Brenton R. Graveley. 2010. <span>“Conservation of an RNA Regulatory Map Between <span><em>Drosophila</em></span> and Mammals.”</span> <em>Genome Research</em> 21 (2): 193–202. <a href="https://doi.org/10.1101/gr.108662.110">https://doi.org/10.1101/gr.108662.110</a>.
</div>
</div></section></div></main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const disableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'prefetch';
    }
  }
  const enableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'stylesheet';
    }
  }
  const manageTransitions = (selector, allowTransitions) => {
    const els = window.document.querySelectorAll(selector);
    for (let i=0; i < els.length; i++) {
      const el = els[i];
      if (allowTransitions) {
        el.classList.remove('notransition');
      } else {
        el.classList.add('notransition');
      }
    }
  }
  const toggleColorMode = (alternate) => {
    // Switch the stylesheets
    const alternateStylesheets = window.document.querySelectorAll('link.quarto-color-scheme.quarto-color-alternate');
    manageTransitions('#quarto-margin-sidebar .nav-link', false);
    if (alternate) {
      enableStylesheet(alternateStylesheets);
      for (const sheetNode of alternateStylesheets) {
        if (sheetNode.id === "quarto-bootstrap") {
          toggleBodyColorMode(sheetNode);
        }
      }
    } else {
      disableStylesheet(alternateStylesheets);
      toggleBodyColorPrimary();
    }
    manageTransitions('#quarto-margin-sidebar .nav-link', true);
    // Switch the toggles
    const toggles = window.document.querySelectorAll('.quarto-color-scheme-toggle');
    for (let i=0; i < toggles.length; i++) {
      const toggle = toggles[i];
      if (toggle) {
        if (alternate) {
          toggle.classList.add("alternate");     
        } else {
          toggle.classList.remove("alternate");
        }
      }
    }
    // Hack to workaround the fact that safari doesn't
    // properly recolor the scrollbar when toggling (#1455)
    if (navigator.userAgent.indexOf('Safari') > 0 && navigator.userAgent.indexOf('Chrome') == -1) {
      manageTransitions("body", false);
      window.scrollTo(0, 1);
      setTimeout(() => {
        window.scrollTo(0, 0);
        manageTransitions("body", true);
      }, 40);  
    }
  }
  const isFileUrl = () => { 
    return window.location.protocol === 'file:';
  }
  const hasAlternateSentinel = () => {  
    let styleSentinel = getColorSchemeSentinel();
    if (styleSentinel !== null) {
      return styleSentinel === "alternate";
    } else {
      return false;
    }
  }
  const setStyleSentinel = (alternate) => {
    const value = alternate ? "alternate" : "default";
    if (!isFileUrl()) {
      window.localStorage.setItem("quarto-color-scheme", value);
    } else {
      localAlternateSentinel = value;
    }
  }
  const getColorSchemeSentinel = () => {
    if (!isFileUrl()) {
      const storageValue = window.localStorage.getItem("quarto-color-scheme");
      return storageValue != null ? storageValue : localAlternateSentinel;
    } else {
      return localAlternateSentinel;
    }
  }
  let localAlternateSentinel = 'default';
  // Dark / light mode switch
  window.quartoToggleColorScheme = () => {
    // Read the current dark / light value 
    let toAlternate = !hasAlternateSentinel();
    toggleColorMode(toAlternate);
    setStyleSentinel(toAlternate);
  };
  // Ensure there is a toggle, if there isn't float one in the top right
  if (window.document.querySelector('.quarto-color-scheme-toggle') === null) {
    const a = window.document.createElement('a');
    a.classList.add('top-right');
    a.classList.add('quarto-color-scheme-toggle');
    a.href = "";
    a.onclick = function() { try { window.quartoToggleColorScheme(); } catch {} return false; };
    const i = window.document.createElement("i");
    i.classList.add('bi');
    a.appendChild(i);
    window.document.body.appendChild(a);
  }
  // Switch to dark mode if need be
  if (hasAlternateSentinel()) {
    toggleColorMode(true);
  } else {
    toggleColorMode(false);
  }
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Handle positioning of the toggle
      window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
      const annoteTargets = window.document.querySelectorAll('.code-annotation-anchor');
      for (let i=0; i<annoteTargets.length; i++) {
        const annoteTarget = annoteTargets[i];
        const targetCell = annoteTarget.getAttribute("data-target-cell");
        const targetAnnotation = annoteTarget.getAttribute("data-target-annotation");
        const contentFn = () => {
          const content = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          if (content) {
            const tipContent = content.cloneNode(true);
            tipContent.classList.add("code-annotation-tip-content");
            return tipContent.outerHTML;
          }
        }
        const config = {
          allowHTML: true,
          content: contentFn,
          onShow: (instance) => {
            selectCodeLines(instance.reference);
            instance.reference.classList.add('code-annotation-active');
            window.tippy.hideAll();
          },
          onHide: (instance) => {
            unselectCodeLines();
            instance.reference.classList.remove('code-annotation-active');
          },
          maxWidth: 300,
          delay: [50, 0],
          duration: [200, 0],
          offset: [5, 10],
          arrow: true,
          appendTo: function(el) {
            return el.parentElement.parentElement.parentElement;
          },
          interactive: true,
          interactiveBorder: 10,
          theme: 'quarto',
          placement: 'right',
          popperOptions: {
            modifiers: [
            {
              name: 'flip',
              options: {
                flipVariations: false, // true by default
                allowedAutoPlacements: ['right'],
                fallbackPlacements: ['right', 'top', 'top-start', 'top-end'],
              },
            },
            {
              name: 'preventOverflow',
              options: {
                mainAxis: false,
                altAxis: false
              }
            }
            ]        
          }      
        };
        window.tippy(annoteTarget, config); 
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left">© Copyright 2023, Jiratchaya Nuanpirom</div>   
    <div class="nav-footer-center">
      &nbsp;
    </div>
    <div class="nav-footer-right">This page is built with ❤️ and <a href="https://quarto.org/">Quarto</a>.</div>
  </div>
</footer>



</body></html>